// @wix/auto_sdk_blog_categories@1.0.1/build/internal/es/src/blog-v3-category-categories.context.js
import { createCategory as publicCreateCategory, updateCategory as publicUpdateCategory, getCategory as publicGetCategory, getCategoryBySlug as publicGetCategoryBySlug, listCategories as publicListCategories, queryCategories as publicQueryCategories, deleteCategory as publicDeleteCategory, } from './blog-v3-category-categories.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onCategoryCreated as publicOnCategoryCreated } from './blog-v3-category-categories.public.js';
import { onCategoryDeleted as publicOnCategoryDeleted } from './blog-v3-category-categories.public.js';
import { onCategoryUpdated as publicOnCategoryUpdated } from './blog-v3-category-categories.public.js';
export const createCategory = /*#__PURE__*/ createRESTModule(publicCreateCategory);
export const updateCategory = /*#__PURE__*/ createRESTModule(publicUpdateCategory);
export const getCategory = /*#__PURE__*/ createRESTModule(publicGetCategory);
export const getCategoryBySlug = /*#__PURE__*/ createRESTModule(publicGetCategoryBySlug);
export const listCategories = /*#__PURE__*/ createRESTModule(publicListCategories);
export const queryCategories = /*#__PURE__*/ createRESTModule(publicQueryCategories);
export const deleteCategory = /*#__PURE__*/ createRESTModule(publicDeleteCategory);
/**
 * Triggered when a category is created.
 */
export const onCategoryCreated = createEventModule(publicOnCategoryCreated);
/**
 * Triggered when a category is deleted.
 */
export const onCategoryDeleted = createEventModule(publicOnCategoryDeleted);
/**
 * Triggered when a category is updated.
 */
export const onCategoryUpdated = createEventModule(publicOnCategoryUpdated);
export { Field, SortOrder, WebhookIdentityType, } from './blog-v3-category-categories.universal.js';
