// @wix/auto_sdk_blog_categories@1.0.1/build/internal/es/src/blog-v3-category-categories.meta.js
import * as ambassadorWixBlogV3Category from './blog-v3-category-categories.http.js';
export function createCategory() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Category.createCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/categories',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateCategory() {
    const payload = { category: { id: ':categoryId' } };
    const getRequestOptions = ambassadorWixBlogV3Category.updateCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/categories/{category.id}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBlogV3Category.getCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getCategoryBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixBlogV3Category.getCategoryBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/categories/slugs/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Category.listCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/categories',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryCategories() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Category.queryCategories(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/categories/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteCategory() {
    const payload = { categoryId: ':categoryId' };
    const getRequestOptions = ambassadorWixBlogV3Category.deleteCategory(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/categories/{categoryId}',
        pathParams: { categoryId: 'categoryId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
