// @wix/auto_sdk_blog_tags/build/es/src/blog-v3-tag-tags.context.js
import { createTag as publicCreateTag, getTagByLabel as publicGetTagByLabel, getTag as publicGetTag, getTagBySlug as publicGetTagBySlug, queryTags as publicQueryTags, deleteTag as publicDeleteTag, } from './blog-v3-tag-tags.public.js';
import { createRESTModule } from '@wix/sdk-runtime/rest-modules';
import { createEventModule } from '@wix/sdk-runtime/event-definition-modules';
import { onTagCreated as publicOnTagCreated } from './blog-v3-tag-tags.public.js';
import { onTagDeleted as publicOnTagDeleted } from './blog-v3-tag-tags.public.js';
import { onTagUpdated as publicOnTagUpdated } from './blog-v3-tag-tags.public.js';
export const createTag = /*#__PURE__*/ createRESTModule(publicCreateTag);
export const getTagByLabel = /*#__PURE__*/ createRESTModule(publicGetTagByLabel);
export const getTag = /*#__PURE__*/ createRESTModule(publicGetTag);
export const getTagBySlug = /*#__PURE__*/ createRESTModule(publicGetTagBySlug);
export const queryTags = /*#__PURE__*/ createRESTModule(publicQueryTags);
export const deleteTag = /*#__PURE__*/ createRESTModule(publicDeleteTag);
/**
 * Triggered when a tag is created.
 */
export const onTagCreated = createEventModule(publicOnTagCreated);
/**
 * Triggered when a tag is deleted.
 */
export const onTagDeleted = createEventModule(publicOnTagDeleted);
/**
 * Triggered when a tag is updated.
 */
export const onTagUpdated = createEventModule(publicOnTagUpdated);
export { Field, SortOrder, WebhookIdentityType, } from './blog-v3-tag-tags.universal.js';
