// @wix/auto_sdk_blog_posts/build/internal/cjs/src/blog-v3-post-posts.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoteRole = exports.ViewRole = exports.ViewMode = exports.VerticalAlignment = exports.Type = exports.ThumbnailsAlignment = exports.TextAlignment = exports.Target = exports.Status = exports.Source = exports.SortOrder = exports.QueryPublicationsCountStatsRequestOrder = exports.PostFieldField = exports.PollLayoutType = exports.PollLayoutDirection = exports.PluginContainerDataAlignment = exports.Origin = exports.Orientation = exports.Order = exports.NullValue = exports.NodeType = exports.ModerationStatusStatus = exports.MapType = exports.LineStyle = exports.LayoutType = exports.InitialExpandedItems = exports.GetPostsSort = exports.GetPostTemplatesSort = exports.GIFType = exports.FontType = exports.Field = exports.Direction = exports.DecorationType = exports.Crop = exports.BackgroundType = exports.AppType = exports.Alignment = exports.onPostUpdated = exports.onPostUnliked = exports.onPostLiked = exports.onPostDeleted = exports.onPostCreated = exports.getTotalPosts = exports.queryPostCountStats = exports.getPostMetrics = exports.queryPosts = exports.listPosts = exports.getPostBySlug = exports.getPost = exports.__metadata = void 0;
exports.WidthType = exports.Width = exports.WebhookIdentityType = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const bytes_1 = require("@wix/sdk-runtime/transformations/bytes");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_1 = require("@wix/sdk-runtime/transformations/page-url");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const video_1 = require("@wix/sdk-runtime/transformations/video");
const video_v2_1 = require("@wix/sdk-runtime/transformations/video-v2");
const sdk_types_1 = require("@wix/sdk-types");
const blog_v3_post_posts_universal_js_1 = require("./blog-v3-post-posts.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function getPost(httpClient) {
    return (postId, options) => (0, blog_v3_post_posts_universal_js_1.getPost)(postId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getPost = getPost;
function getPostBySlug(httpClient) {
    return (slug, options) => (0, blog_v3_post_posts_universal_js_1.getPostBySlug)(slug, options, 
    // @ts-ignore
    { httpClient });
}
exports.getPostBySlug = getPostBySlug;
function listPosts(httpClient) {
    return (options) => (0, blog_v3_post_posts_universal_js_1.listPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listPosts = listPosts;
function queryPosts(httpClient) {
    return (options) => (0, blog_v3_post_posts_universal_js_1.queryPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryPosts = queryPosts;
function getPostMetrics(httpClient) {
    return (postId) => (0, blog_v3_post_posts_universal_js_1.getPostMetrics)(postId, 
    // @ts-ignore
    { httpClient });
}
exports.getPostMetrics = getPostMetrics;
function queryPostCountStats(httpClient) {
    return (options) => (0, blog_v3_post_posts_universal_js_1.queryPostCountStats)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryPostCountStats = queryPostCountStats;
function getTotalPosts(httpClient) {
    return (options) => (0, blog_v3_post_posts_universal_js_1.getTotalPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.getTotalPosts = getTotalPosts;
exports.onPostCreated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.post_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.lastPublishedDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: bytes_1.transformRESTBytesToSDKBytes,
        paths: [{ path: 'entity.richContentCompressed' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.metrics.averageRating' },
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
]), ['entity.rich_content']))();
exports.onPostDeleted = (0, sdk_types_1.EventDefinition)('wix.blog.v3.post_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.lastPublishedDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: bytes_1.transformRESTBytesToSDKBytes,
        paths: [{ path: 'entity.richContentCompressed' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.metrics.averageRating' },
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
]), ['entity.rich_content']))();
exports.onPostLiked = (0, sdk_types_1.EventDefinition)('wix.blog.v3.post_liked', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
]), ['entity.rich_content']))();
exports.onPostUnliked = (0, sdk_types_1.EventDefinition)('wix.blog.v3.post_unliked', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [{ path: 'metadata.eventTime' }],
    },
]), ['entity.rich_content']))();
exports.onPostUpdated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.post_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.lastPublishedDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: bytes_1.transformRESTBytesToSDKBytes,
        paths: [{ path: 'entity.richContentCompressed' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            { path: 'entity.metrics.averageRating' },
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
]), ['entity.rich_content']))();
var blog_v3_post_posts_universal_js_2 = require("./blog-v3-post-posts.universal.js");
Object.defineProperty(exports, "Alignment", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Alignment; } });
Object.defineProperty(exports, "AppType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.AppType; } });
Object.defineProperty(exports, "BackgroundType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.BackgroundType; } });
Object.defineProperty(exports, "Crop", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Crop; } });
Object.defineProperty(exports, "DecorationType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.DecorationType; } });
Object.defineProperty(exports, "Direction", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Direction; } });
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Field; } });
Object.defineProperty(exports, "FontType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.FontType; } });
Object.defineProperty(exports, "GIFType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.GIFType; } });
Object.defineProperty(exports, "GetPostTemplatesSort", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.GetPostTemplatesSort; } });
Object.defineProperty(exports, "GetPostsSort", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.GetPostsSort; } });
Object.defineProperty(exports, "InitialExpandedItems", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.InitialExpandedItems; } });
Object.defineProperty(exports, "LayoutType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.LayoutType; } });
Object.defineProperty(exports, "LineStyle", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.LineStyle; } });
Object.defineProperty(exports, "MapType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.MapType; } });
Object.defineProperty(exports, "ModerationStatusStatus", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.ModerationStatusStatus; } });
Object.defineProperty(exports, "NodeType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.NodeType; } });
Object.defineProperty(exports, "NullValue", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.NullValue; } });
Object.defineProperty(exports, "Order", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Order; } });
Object.defineProperty(exports, "Orientation", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Orientation; } });
Object.defineProperty(exports, "Origin", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Origin; } });
Object.defineProperty(exports, "PluginContainerDataAlignment", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.PluginContainerDataAlignment; } });
Object.defineProperty(exports, "PollLayoutDirection", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.PollLayoutDirection; } });
Object.defineProperty(exports, "PollLayoutType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.PollLayoutType; } });
Object.defineProperty(exports, "PostFieldField", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.PostFieldField; } });
Object.defineProperty(exports, "QueryPublicationsCountStatsRequestOrder", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.QueryPublicationsCountStatsRequestOrder; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Source; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Status; } });
Object.defineProperty(exports, "Target", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Target; } });
Object.defineProperty(exports, "TextAlignment", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.TextAlignment; } });
Object.defineProperty(exports, "ThumbnailsAlignment", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.ThumbnailsAlignment; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Type; } });
Object.defineProperty(exports, "VerticalAlignment", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.VerticalAlignment; } });
Object.defineProperty(exports, "ViewMode", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.ViewMode; } });
Object.defineProperty(exports, "ViewRole", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.ViewRole; } });
Object.defineProperty(exports, "VoteRole", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.VoteRole; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "Width", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.Width; } });
Object.defineProperty(exports, "WidthType", { enumerable: true, get: function () { return blog_v3_post_posts_universal_js_2.WidthType; } });
