// @wix/auto_sdk_blog_blog-importer@1.0.0/build/internal/cjs/src/blog-v1-wordpress-import-blog-importer.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getNotImportedPosts = exports.getImportStatus = exports.deleteNotImported = exports.validateUrlForImport = exports.submitUrlForImport = exports.startImport = void 0;
const ambassadorWixBlogV1WordpressImport = __importStar(require("./blog-v1-wordpress-import-blog-importer.http.js"));
function startImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.startImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.startImport = startImport;
function submitUrlForImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.submitUrlForImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/prepare-wordpress-url-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.submitUrlForImport = submitUrlForImport;
function validateUrlForImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.validateUrlForImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/validate-wordpress-url-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.validateUrlForImport = validateUrlForImport;
function deleteNotImported() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.deleteNotImported(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteNotImported = deleteNotImported;
function getImportStatus() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.getImportStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getImportStatus = getImportStatus;
function getNotImportedPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.getNotImportedPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/wordpress-import/posts/not-imported',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getNotImportedPosts = getNotImportedPosts;
