// @wix/auto_sdk_blog_blog-importer/build/cjs/src/blog-v1-wordpress-import-blog-importer.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getNotImportedPosts = exports.getImportStatus = exports.deleteNotImported = exports.validateUrlForImport = exports.submitUrlForImport = exports.startImport = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixBlogWordpressImportServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/xml-blog-importer',
                destPath: '',
            },
        ],
        _api_base_domain_: [
            {
                srcPath: '/_api/xml-blog-importer',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_blog_blog-importer';
/** Starts wordpress import process. */
function startImport(payload) {
    function __startImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.StartImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __startImport;
}
exports.startImport = startImport;
/**
 * Submits wordpress import process from URL.
 * It will prepare wordpress data for import from the given URL.
 * Use "GetImportStatus" to get the status of the import process.
 * Once the import status becomes READY_TO_IMPORT, the import process can be started by invoking "StartImport".
 */
function submitUrlForImport(payload) {
    function __submitUrlForImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.SubmitUrlForImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/prepare-wordpress-url-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __submitUrlForImport;
}
exports.submitUrlForImport = submitUrlForImport;
function validateUrlForImport(payload) {
    function __validateUrlForImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.ValidateUrlForImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/validate-wordpress-url-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __validateUrlForImport;
}
exports.validateUrlForImport = validateUrlForImport;
function deleteNotImported(payload) {
    function __deleteNotImported({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'DELETE',
            methodFqn: 'com.wix.blog.WordpressImportService.DeleteNotImported',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteNotImported;
}
exports.deleteNotImported = deleteNotImported;
/** Gets the status of the import process. */
function getImportStatus(payload) {
    function __getImportStatus({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'GET',
            methodFqn: 'com.wix.blog.WordpressImportService.GetImportStatus',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getImportStatus;
}
exports.getImportStatus = getImportStatus;
/** Gets the posts that were not yet imported either because of an error or because import is still in progress. */
function getNotImportedPosts(payload) {
    function __getNotImportedPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'GET',
            methodFqn: 'com.wix.blog.WordpressImportService.GetNotImportedPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import/posts/not-imported',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __getNotImportedPosts;
}
exports.getNotImportedPosts = getNotImportedPosts;
