// @wix/auto_sdk_blog_categories/build/internal/cjs/src/blog-v3-category-categories.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Field = void 0;
var Field;
(function (Field) {
    Field["UNKNOWN"] = "UNKNOWN";
    /** Includes Category url. */
    Field["URL"] = "URL";
    /**
     * Includes internal id field.
     * Reserved for internal use
     */
    Field["INTERNAL_ID"] = "INTERNAL_ID";
    /** Includes SEO data. */
    Field["SEO"] = "SEO";
})(Field || (exports.Field = Field = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
