// @wix/auto_sdk_blog_blog-cache@1.0.0/build/cjs/src/blog-v1-blog-cache-blog-cache.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBlogCache = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBlogV1BlogCacheServiceUrl(opts) {
    const domainToMappings = {
        'bo._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'apps._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/blog-cache-service',
                destPath: '',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/blog-cache-service',
                destPath: '',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_blog_blog-cache';
/** Gets blog cache. */
function getBlogCache(payload) {
    function __getBlogCache({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.blog_cache',
            method: 'GET',
            methodFqn: 'wix.blog.v1.BlogCacheService.GetBlogCache',
            packageName: PACKAGE_NAME,
            url: resolveWixBlogV1BlogCacheServiceUrl({
                protoPath: '/v1/blog-cache',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'blogCache.updatedDate' }],
                },
            ]),
        };
        return metadata;
    }
    return __getBlogCache;
}
exports.getBlogCache = getBlogCache;
