// @wix/auto_sdk_blog_blog-cache@1.0.0/build/cjs/src/blog-v1-blog-cache-blog-cache.universal.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBlogCache = exports.DayOfWeek = exports.PlacementType = exports.ResolutionMethod = exports.Flag = exports.LocaleStatus = void 0;
const transform_error_1 = require("@wix/sdk-runtime/transform-error");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const ambassadorWixBlogV1BlogCache = __importStar(require("./blog-v1-blog-cache-blog-cache.http.js"));
var LocaleStatus;
(function (LocaleStatus) {
    /** Value not specified */
    LocaleStatus["UNDEFINED_STATUS"] = "UNDEFINED_STATUS";
    /** Locale was added to site but was not activated */
    LocaleStatus["INACTIVE"] = "INACTIVE";
    /** Locale was added to site and activated */
    LocaleStatus["ACTIVE"] = "ACTIVE";
    /** Default locale (always active) */
    LocaleStatus["DEFAULT"] = "DEFAULT";
})(LocaleStatus || (exports.LocaleStatus = LocaleStatus = {}));
var Flag;
(function (Flag) {
    Flag["UNDEFINED_FLAG_CODE"] = "UNDEFINED_FLAG_CODE";
    Flag["AFG"] = "AFG";
    Flag["AGO"] = "AGO";
    Flag["AIA"] = "AIA";
    Flag["ALA"] = "ALA";
    Flag["ALB"] = "ALB";
    Flag["AND"] = "AND";
    Flag["ARE"] = "ARE";
    Flag["ARG"] = "ARG";
    Flag["ARM"] = "ARM";
    Flag["ASM"] = "ASM";
    Flag["ATF"] = "ATF";
    Flag["ATG"] = "ATG";
    Flag["AUS"] = "AUS";
    Flag["AUT"] = "AUT";
    Flag["AZE"] = "AZE";
    Flag["BDI"] = "BDI";
    Flag["BEL"] = "BEL";
    Flag["BEN"] = "BEN";
    Flag["BES"] = "BES";
    Flag["BFA"] = "BFA";
    Flag["BGD"] = "BGD";
    Flag["BGR"] = "BGR";
    Flag["BHR"] = "BHR";
    Flag["BHS"] = "BHS";
    Flag["BIH"] = "BIH";
    Flag["BLM"] = "BLM";
    Flag["BLR"] = "BLR";
    Flag["BLZ"] = "BLZ";
    Flag["BMU"] = "BMU";
    Flag["BOL"] = "BOL";
    Flag["BRA"] = "BRA";
    Flag["BRB"] = "BRB";
    Flag["BRN"] = "BRN";
    Flag["BRT"] = "BRT";
    Flag["BTN"] = "BTN";
    Flag["BWA"] = "BWA";
    Flag["CAF"] = "CAF";
    Flag["CAN"] = "CAN";
    Flag["CAT"] = "CAT";
    Flag["CHE"] = "CHE";
    Flag["CHL"] = "CHL";
    Flag["CHN"] = "CHN";
    Flag["CIV"] = "CIV";
    Flag["CMR"] = "CMR";
    Flag["COD"] = "COD";
    Flag["COG"] = "COG";
    Flag["COL"] = "COL";
    Flag["COM"] = "COM";
    Flag["COR"] = "COR";
    Flag["CPV"] = "CPV";
    Flag["CRI"] = "CRI";
    Flag["CUB"] = "CUB";
    Flag["CUW"] = "CUW";
    Flag["CYM"] = "CYM";
    Flag["CZE"] = "CZE";
    Flag["DEU"] = "DEU";
    Flag["DJI"] = "DJI";
    Flag["DMA"] = "DMA";
    Flag["DNK"] = "DNK";
    Flag["DOM"] = "DOM";
    Flag["DZA"] = "DZA";
    Flag["ECU"] = "ECU";
    Flag["EGY"] = "EGY";
    Flag["ERI"] = "ERI";
    Flag["ESP"] = "ESP";
    Flag["EST"] = "EST";
    Flag["ETH"] = "ETH";
    Flag["EUR"] = "EUR";
    Flag["EUS"] = "EUS";
    Flag["FIN"] = "FIN";
    Flag["FJI"] = "FJI";
    Flag["FLK"] = "FLK";
    Flag["FRA"] = "FRA";
    Flag["FRO"] = "FRO";
    Flag["FSM"] = "FSM";
    Flag["GAB"] = "GAB";
    Flag["GBR"] = "GBR";
    Flag["GEO"] = "GEO";
    Flag["GGY"] = "GGY";
    Flag["GHA"] = "GHA";
    Flag["GIB"] = "GIB";
    Flag["GIN"] = "GIN";
    Flag["GLG"] = "GLG";
    Flag["GLO"] = "GLO";
    Flag["GLP"] = "GLP";
    Flag["GMB"] = "GMB";
    Flag["GNB"] = "GNB";
    Flag["GNQ"] = "GNQ";
    Flag["GRC"] = "GRC";
    Flag["GRD"] = "GRD";
    Flag["GRL"] = "GRL";
    Flag["GTM"] = "GTM";
    Flag["GUM"] = "GUM";
    Flag["GUY"] = "GUY";
    Flag["HKG"] = "HKG";
    Flag["HND"] = "HND";
    Flag["HRV"] = "HRV";
    Flag["HTI"] = "HTI";
    Flag["HUN"] = "HUN";
    Flag["IDN"] = "IDN";
    Flag["IMN"] = "IMN";
    Flag["IND"] = "IND";
    Flag["IOT"] = "IOT";
    Flag["IRL"] = "IRL";
    Flag["IRN"] = "IRN";
    Flag["IRQ"] = "IRQ";
    Flag["ISL"] = "ISL";
    Flag["ISR"] = "ISR";
    Flag["ITA"] = "ITA";
    Flag["JAM"] = "JAM";
    Flag["JEY"] = "JEY";
    Flag["JOR"] = "JOR";
    Flag["JPN"] = "JPN";
    Flag["KAZ"] = "KAZ";
    Flag["KEN"] = "KEN";
    Flag["KGZ"] = "KGZ";
    Flag["KHM"] = "KHM";
    Flag["KIR"] = "KIR";
    Flag["KNA"] = "KNA";
    Flag["KOR"] = "KOR";
    Flag["KUR"] = "KUR";
    Flag["KWT"] = "KWT";
    Flag["LAO"] = "LAO";
    Flag["LBN"] = "LBN";
    Flag["LBR"] = "LBR";
    Flag["LBY"] = "LBY";
    Flag["LCA"] = "LCA";
    Flag["LIE"] = "LIE";
    Flag["LKA"] = "LKA";
    Flag["LSO"] = "LSO";
    Flag["LTU"] = "LTU";
    Flag["LUX"] = "LUX";
    Flag["LVA"] = "LVA";
    Flag["MAC"] = "MAC";
    Flag["MAF"] = "MAF";
    Flag["MAR"] = "MAR";
    Flag["MCO"] = "MCO";
    Flag["MDA"] = "MDA";
    Flag["MDG"] = "MDG";
    Flag["MDV"] = "MDV";
    Flag["MEX"] = "MEX";
    Flag["MHL"] = "MHL";
    Flag["MKD"] = "MKD";
    Flag["MLI"] = "MLI";
    Flag["MLT"] = "MLT";
    Flag["MMR"] = "MMR";
    Flag["MNE"] = "MNE";
    Flag["MNG"] = "MNG";
    Flag["MNP"] = "MNP";
    Flag["MOZ"] = "MOZ";
    Flag["MRT"] = "MRT";
    Flag["MSR"] = "MSR";
    Flag["MTQ"] = "MTQ";
    Flag["MWI"] = "MWI";
    Flag["MYS"] = "MYS";
    Flag["MYT"] = "MYT";
    Flag["NAM"] = "NAM";
    Flag["NCL"] = "NCL";
    Flag["NER"] = "NER";
    Flag["NFK"] = "NFK";
    Flag["NGA"] = "NGA";
    Flag["NIC"] = "NIC";
    Flag["NIU"] = "NIU";
    Flag["NLD"] = "NLD";
    Flag["NOR"] = "NOR";
    Flag["NPL"] = "NPL";
    Flag["NRU"] = "NRU";
    Flag["NZL"] = "NZL";
    Flag["OMN"] = "OMN";
    Flag["PAK"] = "PAK";
    Flag["PAN"] = "PAN";
    Flag["PCN"] = "PCN";
    Flag["PER"] = "PER";
    Flag["PHL"] = "PHL";
    Flag["PLW"] = "PLW";
    Flag["PNG"] = "PNG";
    Flag["POL"] = "POL";
    Flag["PRI"] = "PRI";
    Flag["PRK"] = "PRK";
    Flag["PRT"] = "PRT";
    Flag["PRY"] = "PRY";
    Flag["PSE"] = "PSE";
    Flag["PYF"] = "PYF";
    Flag["QAT"] = "QAT";
    Flag["REU"] = "REU";
    Flag["ROU"] = "ROU";
    Flag["RUS"] = "RUS";
    Flag["RWA"] = "RWA";
    Flag["SAU"] = "SAU";
    Flag["SDN"] = "SDN";
    Flag["SEN"] = "SEN";
    Flag["SGP"] = "SGP";
    Flag["SGS"] = "SGS";
    Flag["SHN"] = "SHN";
    Flag["SJM"] = "SJM";
    Flag["SLB"] = "SLB";
    Flag["SLE"] = "SLE";
    Flag["SLV"] = "SLV";
    Flag["SMR"] = "SMR";
    Flag["SOM"] = "SOM";
    Flag["SPM"] = "SPM";
    Flag["SRB"] = "SRB";
    Flag["SSD"] = "SSD";
    Flag["STP"] = "STP";
    Flag["SUR"] = "SUR";
    Flag["SVK"] = "SVK";
    Flag["SVN"] = "SVN";
    Flag["SWE"] = "SWE";
    Flag["SWZ"] = "SWZ";
    Flag["SXM"] = "SXM";
    Flag["SYC"] = "SYC";
    Flag["SYR"] = "SYR";
    Flag["TCA"] = "TCA";
    Flag["TCD"] = "TCD";
    Flag["TGO"] = "TGO";
    Flag["THA"] = "THA";
    Flag["TJK"] = "TJK";
    Flag["TKL"] = "TKL";
    Flag["TKM"] = "TKM";
    Flag["TLS"] = "TLS";
    Flag["TON"] = "TON";
    Flag["TTO"] = "TTO";
    Flag["TUN"] = "TUN";
    Flag["TUR"] = "TUR";
    Flag["TUV"] = "TUV";
    Flag["TWN"] = "TWN";
    Flag["TZA"] = "TZA";
    Flag["UGA"] = "UGA";
    Flag["UKR"] = "UKR";
    Flag["URY"] = "URY";
    Flag["USA"] = "USA";
    Flag["UZB"] = "UZB";
    Flag["VAT"] = "VAT";
    Flag["VCT"] = "VCT";
    Flag["VEN"] = "VEN";
    Flag["VGB"] = "VGB";
    Flag["VIR"] = "VIR";
    Flag["VLC"] = "VLC";
    Flag["VNM"] = "VNM";
    Flag["VUT"] = "VUT";
    Flag["WLF"] = "WLF";
    Flag["WLS"] = "WLS";
    Flag["YEM"] = "YEM";
    Flag["ZAF"] = "ZAF";
    Flag["ZMB"] = "ZMB";
    Flag["ZWE"] = "ZWE";
})(Flag || (exports.Flag = Flag = {}));
var ResolutionMethod;
(function (ResolutionMethod) {
    ResolutionMethod["QUERY_PARAM"] = "QUERY_PARAM";
    ResolutionMethod["SUBDOMAIN"] = "SUBDOMAIN";
    ResolutionMethod["SUBDIRECTORY"] = "SUBDIRECTORY";
})(ResolutionMethod || (exports.ResolutionMethod = ResolutionMethod = {}));
/** Where the extra text should be displayed: before, after or instead of the actual address. */
var PlacementType;
(function (PlacementType) {
    PlacementType["BEFORE"] = "BEFORE";
    PlacementType["AFTER"] = "AFTER";
    PlacementType["REPLACE"] = "REPLACE";
})(PlacementType || (exports.PlacementType = PlacementType = {}));
/** Enumerates the days of the week. */
var DayOfWeek;
(function (DayOfWeek) {
    DayOfWeek["MONDAY"] = "MONDAY";
    DayOfWeek["TUESDAY"] = "TUESDAY";
    DayOfWeek["WEDNESDAY"] = "WEDNESDAY";
    DayOfWeek["THURSDAY"] = "THURSDAY";
    DayOfWeek["FRIDAY"] = "FRIDAY";
    DayOfWeek["SATURDAY"] = "SATURDAY";
    DayOfWeek["SUNDAY"] = "SUNDAY";
})(DayOfWeek || (exports.DayOfWeek = DayOfWeek = {}));
/**
 * Gets blog cache.
 * @public
 * @documentationMaturity preview
 * @permissionId BLOG.READ_SETTINGS
 * @applicableIdentity VISITOR
 * @fqn wix.blog.v1.BlogCacheService.GetBlogCache
 */
async function getBlogCache() {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[0];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({});
    const reqOpts = ambassadorWixBlogV1BlogCache.getBlogCache(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {},
            singleArgumentUnchanged: false,
        }, []);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.getBlogCache = getBlogCache;
