// @wix/auto_sdk_blog_categories@1.0.1/build/cjs/src/blog-v3-category-categories.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Field = exports.onCategoryUpdated = exports.onCategoryDeleted = exports.onCategoryCreated = exports.deleteCategory = exports.queryCategories = exports.listCategories = exports.getCategoryBySlug = exports.getCategory = exports.updateCategory = exports.createCategory = void 0;
const blog_v3_category_categories_public_js_1 = require("./blog-v3-category-categories.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const blog_v3_category_categories_public_js_2 = require("./blog-v3-category-categories.public.js");
const blog_v3_category_categories_public_js_3 = require("./blog-v3-category-categories.public.js");
const blog_v3_category_categories_public_js_4 = require("./blog-v3-category-categories.public.js");
exports.createCategory = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.createCategory);
exports.updateCategory = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.updateCategory);
exports.getCategory = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.getCategory);
exports.getCategoryBySlug = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.getCategoryBySlug);
exports.listCategories = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.listCategories);
exports.queryCategories = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.queryCategories);
exports.deleteCategory = (0, rest_modules_1.createRESTModule)(blog_v3_category_categories_public_js_1.deleteCategory);
/**
 * Triggered when a category is created.
 */
exports.onCategoryCreated = (0, event_definition_modules_1.createEventModule)(blog_v3_category_categories_public_js_2.onCategoryCreated);
/**
 * Triggered when a category is deleted.
 */
exports.onCategoryDeleted = (0, event_definition_modules_1.createEventModule)(blog_v3_category_categories_public_js_3.onCategoryDeleted);
/**
 * Triggered when a category is updated.
 */
exports.onCategoryUpdated = (0, event_definition_modules_1.createEventModule)(blog_v3_category_categories_public_js_4.onCategoryUpdated);
var blog_v3_category_categories_universal_js_1 = require("./blog-v3-category-categories.universal.js");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_1.Field; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_1.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_1.WebhookIdentityType; } });
