// @wix/auto_sdk_blog_draft-posts@1.0.2/build/internal/es/src/blog-v3-draft-draft-posts.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTFloatToSDKFloat } from '@wix/sdk-runtime/transformations/float';
import { transformRESTImageToSDKImage } from '@wix/sdk-runtime/transformations/image';
import { transformRESTPageURLToSDKPageURL } from '@wix/sdk-runtime/transformations/page-url';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { transformRESTVideoToSDKVideo } from '@wix/sdk-runtime/transformations/video';
import { transformRESTVideoV2ToSDKVideoV2 } from '@wix/sdk-runtime/transformations/video-v2';
import { EventDefinition } from '@wix/sdk-types';
import { bulkCreateDraftPosts as universalBulkCreateDraftPosts, bulkDeleteDraftPosts as universalBulkDeleteDraftPosts, bulkUpdateDraftPosts as universalBulkUpdateDraftPosts, createDraftPost as universalCreateDraftPost, deleteDraftPost as universalDeleteDraftPost, getDeletedDraftPost as universalGetDeletedDraftPost, getDraftPost as universalGetDraftPost, listDeletedDraftPosts as universalListDeletedDraftPosts, listDraftPosts as universalListDraftPosts, publishDraftPost as universalPublishDraftPost, queryDraftPosts as universalQueryDraftPosts, removeFromTrashBin as universalRemoveFromTrashBin, restoreFromTrashBin as universalRestoreFromTrashBin, updateDraftPost as universalUpdateDraftPost, } from './blog-v3-draft-draft-posts.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/blog' };
export function createDraftPost(httpClient) {
    return (draftPost, options) => universalCreateDraftPost(draftPost, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkCreateDraftPosts(httpClient) {
    return (draftPosts, options) => universalBulkCreateDraftPosts(draftPosts, options, 
    // @ts-ignore
    { httpClient });
}
export function bulkUpdateDraftPosts(httpClient) {
    return (options) => universalBulkUpdateDraftPosts(options, 
    // @ts-ignore
    { httpClient });
}
export function listDeletedDraftPosts(httpClient) {
    return (options) => universalListDeletedDraftPosts(options, 
    // @ts-ignore
    { httpClient });
}
export function getDraftPost(httpClient) {
    return (draftPostId, options) => universalGetDraftPost(draftPostId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateDraftPost(httpClient) {
    return (_id, draftPost, options) => universalUpdateDraftPost(_id, draftPost, options, 
    // @ts-ignore
    { httpClient });
}
export function deleteDraftPost(httpClient) {
    return (draftPostId, options) => universalDeleteDraftPost(draftPostId, options, 
    // @ts-ignore
    { httpClient });
}
export function removeFromTrashBin(httpClient) {
    return (draftPostId) => universalRemoveFromTrashBin(draftPostId, 
    // @ts-ignore
    { httpClient });
}
export function bulkDeleteDraftPosts(httpClient) {
    return (postIds, options) => universalBulkDeleteDraftPosts(postIds, options, 
    // @ts-ignore
    { httpClient });
}
export function listDraftPosts(httpClient) {
    return (options) => universalListDraftPosts(options, 
    // @ts-ignore
    { httpClient });
}
export function getDeletedDraftPost(httpClient) {
    return (draftPostId) => universalGetDeletedDraftPost(draftPostId, 
    // @ts-ignore
    { httpClient });
}
export function restoreFromTrashBin(httpClient) {
    return (draftPostId) => universalRestoreFromTrashBin(draftPostId, 
    // @ts-ignore
    { httpClient });
}
export function queryDraftPosts(httpClient) {
    return (options) => universalQueryDraftPosts(options, 
    // @ts-ignore
    { httpClient });
}
export function publishDraftPost(httpClient) {
    return (draftPostId) => universalPublishDraftPost(draftPostId, 
    // @ts-ignore
    { httpClient });
}
export const onDraftCreated = EventDefinition('wix.blog.v3.draft_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
export const onDraftDeleted = EventDefinition('wix.blog.v3.draft_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
export const onDraftUpdated = EventDefinition('wix.blog.v3.draft_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
export { Action, Alignment, AppType, BackgroundType, ButtonDataType, Crop, DecorationType, Direction, Field, FontType, GIFType, GetDraftPostsSort, InitialExpandedItems, LayoutType, LineStyle, MapType, ModerationStatusStatus, NodeType, NullValue, Orientation, Origin, PluginContainerDataAlignment, PollLayoutDirection, PollLayoutType, SortOrder, Source, Status, Target, TextAlignment, ThumbnailsAlignment, TotalDraftPostsGroupingField, Type, VerticalAlignment, ViewMode, ViewRole, VoteRole, WebhookIdentityType, Width, WidthType, } from './blog-v3-draft-draft-posts.universal.js';
