// @wix/auto_sdk_blog_blog-importer@1.0.0/build/es/src/blog-v1-wordpress-import-blog-importer.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveComWixBlogWordpressImportServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/xml-blog-importer',
                destPath: '',
            },
        ],
        _api_base_domain_: [
            {
                srcPath: '/_api/xml-blog-importer',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_blog_blog-importer';
/** Starts wordpress import process. */
export function startImport(payload) {
    function __startImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.StartImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __startImport;
}
/**
 * Submits wordpress import process from URL.
 * It will prepare wordpress data for import from the given URL.
 * Use "GetImportStatus" to get the status of the import process.
 * Once the import status becomes READY_TO_IMPORT, the import process can be started by invoking "StartImport".
 */
export function submitUrlForImport(payload) {
    function __submitUrlForImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.SubmitUrlForImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/prepare-wordpress-url-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __submitUrlForImport;
}
export function validateUrlForImport(payload) {
    function __validateUrlForImport({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'POST',
            methodFqn: 'com.wix.blog.WordpressImportService.ValidateUrlForImport',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/validate-wordpress-url-import',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __validateUrlForImport;
}
export function deleteNotImported(payload) {
    function __deleteNotImported({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'DELETE',
            methodFqn: 'com.wix.blog.WordpressImportService.DeleteNotImported',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __deleteNotImported;
}
/** Gets the status of the import process. */
export function getImportStatus(payload) {
    function __getImportStatus({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'GET',
            methodFqn: 'com.wix.blog.WordpressImportService.GetImportStatus',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __getImportStatus;
}
/** Gets the posts that were not yet imported either because of an error or because import is still in progress. */
export function getNotImportedPosts(payload) {
    function __getNotImportedPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v1.wordpress_import',
            method: 'GET',
            methodFqn: 'com.wix.blog.WordpressImportService.GetNotImportedPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixBlogWordpressImportServiceUrl({
                protoPath: '/v1/wordpress-import/posts/not-imported',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __getNotImportedPosts;
}
