// @wix/blog_app-extensions/build/cjs/src/extensions/dashboard-page-builders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.blogSettings = exports.blogOverview = exports.blogCategories = exports.blogTemplates = exports.blogMonetization = exports.blogComments = exports.blogTags = exports.blogWriters = exports.blogPosts = exports.blogAnalytics = void 0;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Analytics page in the dashboard.
 * @dashobardPageBuilder
 */
function blogAnalytics() {
    return { pageId: '3272eb63-13c2-4e22-bf9f-66964d1d39a3' };
}
exports.blogAnalytics = blogAnalytics;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Posts page in the dashboard.
 * @dashobardPageBuilder
 */
function blogPosts() {
    return { pageId: 'c7b86028-b1d2-42b0-be1d-4c95b374fc7d' };
}
exports.blogPosts = blogPosts;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Writers page in the dashboard.
 * @dashobardPageBuilder
 */
function blogWriters() {
    return { pageId: 'b8c96d90-95d6-57de-9a97-0d44c4dfaa96' };
}
exports.blogWriters = blogWriters;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Tags page in the dashboard.
 * @dashobardPageBuilder
 */
function blogTags() {
    return { pageId: '06313e49-80f7-498e-99cd-312eab83f391' };
}
exports.blogTags = blogTags;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Comments page in the dashboard.
 * @dashobardPageBuilder
 */
function blogComments() {
    return { pageId: '77f77e2b-56b6-4fe5-8011-ed1ebfd95c0b' };
}
exports.blogComments = blogComments;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Monetization page in the dashboard.
 * @dashobardPageBuilder
 */
function blogMonetization() {
    return { pageId: '681cbbb9-800c-414f-856a-ad4be4e1b296' };
}
exports.blogMonetization = blogMonetization;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Templates page in the dashboard.
 * @dashobardPageBuilder
 */
function blogTemplates() {
    return { pageId: '374b3e95-6860-4018-906b-211aa786450d' };
}
exports.blogTemplates = blogTemplates;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Categories page in the dashboard.
 * @dashobardPageBuilder
 */
function blogCategories() {
    return { pageId: 'fd3e1c66-89b0-44ee-b824-1a28cc1a6dde' };
}
exports.blogCategories = blogCategories;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Overview page in the dashboard.
 * @dashobardPageBuilder
 */
function blogOverview() {
    return { pageId: '7c8f91b9-0b65-47ee-92eb-37c1f9a4d410' };
}
exports.blogOverview = blogOverview;
/**
 * A builder method that returns the <a href='https://dev.wix.com/docs/sdk/host-modules/dashboard/navigate#destination-object'>destination object</a> for navigating to the Blog Settings page in the dashboard.
 * @dashobardPageBuilder
 */
function blogSettings() {
    return { pageId: '2718bafd-5769-487e-b162-4b4d747edf89' };
}
exports.blogSettings = blogSettings;
