// @wix/auto_sdk_blog_categories@1.0.1/build/internal/cjs/src/blog-v3-category-categories.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Field = exports.onCategoryUpdated = exports.onCategoryDeleted = exports.onCategoryCreated = exports.deleteCategory = exports.queryCategories = exports.listCategories = exports.getCategoryBySlug = exports.getCategory = exports.updateCategory = exports.createCategory = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_1 = require("@wix/sdk-runtime/transformations/page-url");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const blog_v3_category_categories_universal_js_1 = require("./blog-v3-category-categories.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function createCategory(httpClient) {
    return (category, options) => (0, blog_v3_category_categories_universal_js_1.createCategory)(category, options, 
    // @ts-ignore
    { httpClient });
}
exports.createCategory = createCategory;
function updateCategory(httpClient) {
    return (_id, category, options) => (0, blog_v3_category_categories_universal_js_1.updateCategory)(_id, category, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateCategory = updateCategory;
function getCategory(httpClient) {
    return (categoryId, options) => (0, blog_v3_category_categories_universal_js_1.getCategory)(categoryId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getCategory = getCategory;
function getCategoryBySlug(httpClient) {
    return (slug, options) => (0, blog_v3_category_categories_universal_js_1.getCategoryBySlug)(slug, options, 
    // @ts-ignore
    { httpClient });
}
exports.getCategoryBySlug = getCategoryBySlug;
function listCategories(httpClient) {
    return (options) => (0, blog_v3_category_categories_universal_js_1.listCategories)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listCategories = listCategories;
function queryCategories(httpClient) {
    return (options) => (0, blog_v3_category_categories_universal_js_1.queryCategories)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryCategories = queryCategories;
function deleteCategory(httpClient) {
    return (categoryId) => (0, blog_v3_category_categories_universal_js_1.deleteCategory)(categoryId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteCategory = deleteCategory;
exports.onCategoryCreated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.category_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.coverImage' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCategoryDeleted = (0, sdk_types_1.EventDefinition)('wix.blog.v3.category_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [
            { path: 'undefined.url' },
            { path: 'undefined.translations.url' },
        ],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'undefined.coverImage' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onCategoryUpdated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.category_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [{ path: 'entity.coverImage' }],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var blog_v3_category_categories_universal_js_2 = require("./blog-v3-category-categories.universal.js");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_2.Field; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_category_categories_universal_js_2.WebhookIdentityType; } });
