// @wix/auto_sdk_blog_posts/build/internal/es/src/blog-v3-post-posts.meta.js
import * as ambassadorWixBlogV3Post from './blog-v3-post-posts.http.js';
export function getPost() {
    const payload = { postId: ':postId' };
    const getRequestOptions = ambassadorWixBlogV3Post.getPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/posts/{postId}',
        pathParams: { postId: 'postId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPostBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixBlogV3Post.getPostBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/posts/slugs/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Post.listPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Post.queryPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/posts/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getPostMetrics() {
    const payload = { postId: ':postId' };
    const getRequestOptions = ambassadorWixBlogV3Post.getPostMetrics(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/posts/{postId}/metrics',
        pathParams: { postId: 'postId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryPostCountStats() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Post.queryPostCountStats(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/stats/post/count',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTotalPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Post.getTotalPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/stats/posts/total',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
