// @wix/auto_sdk_blog_tags/build/internal/cjs/src/blog-v3-tag-tags.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteTag = exports.queryTags = exports.getTagBySlug = exports.getTag = exports.getTagByLabel = exports.createTag = void 0;
const ambassadorWixBlogV3Tag = __importStar(require("./blog-v3-tag-tags.http.js"));
function createTag() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Tag.createTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTag = createTag;
function getTagByLabel() {
    const payload = { label: ':label' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTagByLabel(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/labels/{label}',
        pathParams: { label: 'label' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTagByLabel = getTagByLabel;
function getTag() {
    const payload = { tagId: ':tagId' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTag = getTag;
function getTagBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTagBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/slugs/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTagBySlug = getTagBySlug;
function queryTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Tag.queryTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/tags/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryTags = queryTags;
function deleteTag() {
    const payload = { tagId: ':tagId' };
    const getRequestOptions = ambassadorWixBlogV3Tag.deleteTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteTag = deleteTag;
