// @wix/auto_sdk_blog_tags/build/internal/cjs/src/blog-v3-tag-tags.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Field = exports.onTagUpdated = exports.onTagDeleted = exports.onTagCreated = exports.deleteTag = exports.queryTags = exports.getTagBySlug = exports.getTag = exports.getTagByLabel = exports.createTag = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const page_url_1 = require("@wix/sdk-runtime/transformations/page-url");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const blog_v3_tag_tags_universal_js_1 = require("./blog-v3-tag-tags.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function createTag(httpClient) {
    return (label, options) => (0, blog_v3_tag_tags_universal_js_1.createTag)(label, options, 
    // @ts-ignore
    { httpClient });
}
exports.createTag = createTag;
function getTagByLabel(httpClient) {
    return (label, options) => (0, blog_v3_tag_tags_universal_js_1.getTagByLabel)(label, options, 
    // @ts-ignore
    { httpClient });
}
exports.getTagByLabel = getTagByLabel;
function getTag(httpClient) {
    return (tagId, options) => (0, blog_v3_tag_tags_universal_js_1.getTag)(tagId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getTag = getTag;
function getTagBySlug(httpClient) {
    return (slug, options) => (0, blog_v3_tag_tags_universal_js_1.getTagBySlug)(slug, options, 
    // @ts-ignore
    { httpClient });
}
exports.getTagBySlug = getTagBySlug;
function queryTags(httpClient) {
    return (options) => (0, blog_v3_tag_tags_universal_js_1.queryTags)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryTags = queryTags;
function deleteTag(httpClient) {
    return (tagId) => (0, blog_v3_tag_tags_universal_js_1.deleteTag)(tagId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteTag = deleteTag;
exports.onTagCreated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.tag_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }],
    },
])))();
exports.onTagDeleted = (0, sdk_types_1.EventDefinition)('wix.blog.v3.tag_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'undefined.url' }],
    },
])))();
exports.onTagUpdated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.tag_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }],
    },
])))();
var blog_v3_tag_tags_universal_js_2 = require("./blog-v3-tag-tags.universal.js");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_2.Field; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_2.WebhookIdentityType; } });
