// @wix/auto_sdk_blog_blog-importer/build/internal/cjs/src/blog-v1-wordpress-import-blog-importer.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Status = exports.getNotImportedPosts = exports.getImportStatus = exports.deleteNotImported = exports.validateUrlForImport = exports.submitUrlForImport = exports.startImport = exports.__metadata = void 0;
const blog_v1_wordpress_import_blog_importer_universal_js_1 = require("./blog-v1-wordpress-import-blog-importer.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function startImport(httpClient) {
    return (options) => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.startImport)(options, 
    // @ts-ignore
    { httpClient });
}
exports.startImport = startImport;
function submitUrlForImport(httpClient) {
    return (url, options) => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.submitUrlForImport)(url, options, 
    // @ts-ignore
    { httpClient });
}
exports.submitUrlForImport = submitUrlForImport;
function validateUrlForImport(httpClient) {
    return (url) => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.validateUrlForImport)(url, 
    // @ts-ignore
    { httpClient });
}
exports.validateUrlForImport = validateUrlForImport;
function deleteNotImported(httpClient) {
    return () => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.deleteNotImported)(
    // @ts-ignore
    { httpClient });
}
exports.deleteNotImported = deleteNotImported;
function getImportStatus(httpClient) {
    return () => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.getImportStatus)(
    // @ts-ignore
    { httpClient });
}
exports.getImportStatus = getImportStatus;
function getNotImportedPosts(httpClient) {
    return () => (0, blog_v1_wordpress_import_blog_importer_universal_js_1.getNotImportedPosts)(
    // @ts-ignore
    { httpClient });
}
exports.getNotImportedPosts = getNotImportedPosts;
var blog_v1_wordpress_import_blog_importer_universal_js_2 = require("./blog-v1-wordpress-import-blog-importer.universal.js");
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return blog_v1_wordpress_import_blog_importer_universal_js_2.Status; } });
