// @wix/auto_sdk_blog_draft-posts/build/internal/cjs/src/blog-v3-draft-draft-posts.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.publishDraftPost = exports.queryDraftPosts = exports.restoreFromTrashBin = exports.getDeletedDraftPost = exports.listDraftPosts = exports.bulkDeleteDraftPosts = exports.removeFromTrashBin = exports.deleteDraftPost = exports.updateDraftPost = exports.getDraftPost = exports.listDeletedDraftPosts = exports.bulkUpdateDraftPosts = exports.bulkCreateDraftPosts = exports.createDraftPost = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const float_2 = require("@wix/sdk-runtime/transformations/float");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const field_mask_1 = require("@wix/sdk-runtime/transformations/field-mask");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(opts) {
    const domainToMappings = {
        _api_base_domain_: [
            {
                srcPath: '/draft-post-service-api-proxy',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        '*.pub.wix-code.com': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        _: [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'apps._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'bo._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'wixbo.ai': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'wix-bo.com': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'editor.wixapps.net': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
            {
                srcPath: '/_api/draft-post-service-api-proxy/v3/bulk/draft-posts',
                destPath: '/v3/bulk/draft-posts',
            },
        ],
        'social-blog._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/blog/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
            {
                srcPath: '/blog/v3/bulk/draft-posts/create',
                destPath: '/v3/bulk/draft-posts/create',
            },
            {
                srcPath: '/blog/v3/draft-posts/',
                destPath: '/v3/draft-posts',
            },
            {
                srcPath: '/blog/v3/bulk/draft-posts',
                destPath: '/v3/bulk/draft-posts',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/communities-blog-node-api/v3/draft-posts',
                destPath: '/v3/draft-posts',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_blog_draft-posts';
/**
 * Creates a draft post.
 *
 * The draft post's `memberId` is required for third-party apps.
 */
function createDraftPost(payload) {
    function __createDraftPost({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'draftPost.editedDate' },
                    { path: 'draftPost.scheduledPublishDate' },
                    { path: 'draftPost.firstPublishedDate' },
                    { path: 'draftPost.createdDate' },
                    { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                    { path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate' },
                    { path: 'draftPost.heroImage.urlExpirationDate' },
                    { path: 'draftPost.richContent.metadata.createdTimestamp' },
                    { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                    { path: 'draftPost.moderationDetails.submittedDate' },
                    { path: 'draftPost.moderationDetails.moderationDate' },
                    { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                    { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                    },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'draftPost.coverMedia.image.focalPoint.x' },
                    { path: 'draftPost.coverMedia.image.focalPoint.y' },
                    { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                    { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                    { path: 'draftPost.heroImage.focalPoint.x' },
                    { path: 'draftPost.heroImage.focalPoint.y' },
                    { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                    { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                    },
                    { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                    { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                    { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                    { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                    { path: 'draftPost.richContent.nodes.pollData.poll.image.duration' },
                    {
                        path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'draftPost.richContent.nodes.appEmbedData.image.duration' },
                    { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                    { path: 'draftPost.richContent.nodes.videoData.thumbnail.duration' },
                    { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                    { path: 'draftPost.richContent.nodes.audioData.coverImage.duration' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'POST',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.CreateDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts', data: serializedData, host }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPost.editedDate' },
                        { path: 'draftPost.scheduledPublishDate' },
                        { path: 'draftPost.firstPublishedDate' },
                        { path: 'draftPost.createdDate' },
                        { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPost.heroImage.urlExpirationDate' },
                        { path: 'draftPost.richContent.metadata.createdTimestamp' },
                        { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPost.moderationDetails.submittedDate' },
                        { path: 'draftPost.moderationDetails.moderationDate' },
                        { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPost.coverMedia.image.focalPoint.x' },
                        { path: 'draftPost.coverMedia.image.focalPoint.y' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPost.heroImage.focalPoint.x' },
                        { path: 'draftPost.heroImage.focalPoint.y' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPost.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPost.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createDraftPost;
}
exports.createDraftPost = createDraftPost;
/** Creates multiple draft posts. */
function bulkCreateDraftPosts(payload) {
    function __bulkCreateDraftPosts({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'draftPosts.editedDate' },
                    { path: 'draftPosts.scheduledPublishDate' },
                    { path: 'draftPosts.firstPublishedDate' },
                    { path: 'draftPosts.createdDate' },
                    { path: 'draftPosts.coverMedia.image.urlExpirationDate' },
                    { path: 'draftPosts.coverMedia.video.thumbnail.urlExpirationDate' },
                    { path: 'draftPosts.heroImage.urlExpirationDate' },
                    { path: 'draftPosts.richContent.metadata.createdTimestamp' },
                    { path: 'draftPosts.richContent.metadata.updatedTimestamp' },
                    { path: 'draftPosts.moderationDetails.submittedDate' },
                    { path: 'draftPosts.moderationDetails.moderationDate' },
                    { path: 'draftPosts.media.wixMedia.image.urlExpirationDate' },
                    { path: 'draftPosts.media.wixMedia.videoV2.urlExpirationDate' },
                    {
                        path: 'draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate',
                    },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'draftPosts.coverMedia.image.focalPoint.x' },
                    { path: 'draftPosts.coverMedia.image.focalPoint.y' },
                    { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.x' },
                    { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.y' },
                    { path: 'draftPosts.heroImage.focalPoint.x' },
                    { path: 'draftPosts.heroImage.focalPoint.y' },
                    { path: 'draftPosts.media.wixMedia.image.focalPoint.x' },
                    { path: 'draftPosts.media.wixMedia.image.focalPoint.y' },
                    {
                        path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                    },
                    {
                        path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                    },
                    { path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.x' },
                    { path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.y' },
                    {
                        path: 'draftPosts.richContent.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'draftPosts.richContent.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'draftPosts.richContent.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'draftPosts.richContent.nodes.imageData.image.duration' },
                    { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lat' },
                    { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lng' },
                    { path: 'draftPosts.richContent.nodes.pollData.poll.image.duration' },
                    {
                        path: 'draftPosts.richContent.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'draftPosts.richContent.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'draftPosts.richContent.nodes.appEmbedData.image.duration' },
                    { path: 'draftPosts.richContent.nodes.videoData.video.duration' },
                    { path: 'draftPosts.richContent.nodes.videoData.thumbnail.duration' },
                    { path: 'draftPosts.richContent.nodes.audioData.audio.duration' },
                    {
                        path: 'draftPosts.richContent.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'POST',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkCreateDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/bulk/draft-posts/create', data: serializedData, host }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.editedDate' },
                        { path: 'results.item.scheduledPublishDate' },
                        { path: 'results.item.firstPublishedDate' },
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'results.item.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'results.item.heroImage.urlExpirationDate' },
                        { path: 'results.item.richContent.metadata.createdTimestamp' },
                        { path: 'results.item.richContent.metadata.updatedTimestamp' },
                        { path: 'results.item.moderationDetails.submittedDate' },
                        { path: 'results.item.moderationDetails.moderationDate' },
                        { path: 'results.item.media.wixMedia.image.urlExpirationDate' },
                        { path: 'results.item.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'results.item.coverMedia.image.focalPoint.x' },
                        { path: 'results.item.coverMedia.image.focalPoint.y' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'results.item.heroImage.focalPoint.x' },
                        { path: 'results.item.heroImage.focalPoint.y' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.x' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.richContent.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkCreateDraftPosts;
}
exports.bulkCreateDraftPosts = bulkCreateDraftPosts;
/** Updates multiple draft posts. */
function bulkUpdateDraftPosts(payload) {
    function __bulkUpdateDraftPosts({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'scheduledPublishDate' },
                    { path: 'draftPosts.draftPost.editedDate' },
                    { path: 'draftPosts.draftPost.scheduledPublishDate' },
                    { path: 'draftPosts.draftPost.firstPublishedDate' },
                    { path: 'draftPosts.draftPost.createdDate' },
                    { path: 'draftPosts.draftPost.coverMedia.image.urlExpirationDate' },
                    {
                        path: 'draftPosts.draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                    },
                    { path: 'draftPosts.draftPost.heroImage.urlExpirationDate' },
                    {
                        path: 'draftPosts.draftPost.richContent.metadata.createdTimestamp',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.metadata.updatedTimestamp',
                    },
                    { path: 'draftPosts.draftPost.moderationDetails.submittedDate' },
                    { path: 'draftPosts.draftPost.moderationDetails.moderationDate' },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.image.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                    },
                ],
            },
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'draftPosts.fieldMask' }],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'draftPosts.draftPost.coverMedia.image.focalPoint.x' },
                    { path: 'draftPosts.draftPost.coverMedia.image.focalPoint.y' },
                    {
                        path: 'draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.x',
                    },
                    {
                        path: 'draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.y',
                    },
                    { path: 'draftPosts.draftPost.heroImage.focalPoint.x' },
                    { path: 'draftPosts.draftPost.heroImage.focalPoint.y' },
                    { path: 'draftPosts.draftPost.media.wixMedia.image.focalPoint.x' },
                    { path: 'draftPosts.draftPost.media.wixMedia.image.focalPoint.y' },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.x',
                    },
                    {
                        path: 'draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.y',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.galleryData.options.item.ratio',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.imageData.image.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lat',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lng',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.pollData.poll.image.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.appEmbedData.image.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.videoData.video.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.videoData.thumbnail.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.audioData.audio.duration',
                    },
                    {
                        path: 'draftPosts.draftPost.richContent.nodes.audioData.coverImage.duration',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'PATCH',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkUpdateDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts/update', data: serializedData, host }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.editedDate' },
                        { path: 'results.item.scheduledPublishDate' },
                        { path: 'results.item.firstPublishedDate' },
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'results.item.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'results.item.heroImage.urlExpirationDate' },
                        { path: 'results.item.richContent.metadata.createdTimestamp' },
                        { path: 'results.item.richContent.metadata.updatedTimestamp' },
                        { path: 'results.item.moderationDetails.submittedDate' },
                        { path: 'results.item.moderationDetails.moderationDate' },
                        { path: 'results.item.media.wixMedia.image.urlExpirationDate' },
                        { path: 'results.item.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'results.item.coverMedia.image.focalPoint.x' },
                        { path: 'results.item.coverMedia.image.focalPoint.y' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'results.item.heroImage.focalPoint.x' },
                        { path: 'results.item.heroImage.focalPoint.y' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.x' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.richContent.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkUpdateDraftPosts;
}
exports.bulkUpdateDraftPosts = bulkUpdateDraftPosts;
/**
 * Retrieves a list of up to 100 deleted draft posts.
 *
 * List Deleted Draft Posts runs with these defaults, which you can override:
 * - `editedDate` is sorted in descending order. In this case,`editedDate` implies the date the post was deleted.
 * - `paging.limit` is `50`.
 * - `paging.offset` is `0`.
 */
function listDeletedDraftPosts(payload) {
    function __listDeletedDraftPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'GET',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDeletedDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts/trash-bin', data: payload, host }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPosts.editedDate' },
                        { path: 'draftPosts.scheduledPublishDate' },
                        { path: 'draftPosts.firstPublishedDate' },
                        { path: 'draftPosts.createdDate' },
                        { path: 'draftPosts.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPosts.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPosts.heroImage.urlExpirationDate' },
                        { path: 'draftPosts.richContent.metadata.createdTimestamp' },
                        { path: 'draftPosts.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPosts.moderationDetails.submittedDate' },
                        { path: 'draftPosts.moderationDetails.moderationDate' },
                        { path: 'draftPosts.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPosts.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPosts.coverMedia.image.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.image.focalPoint.y' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPosts.heroImage.focalPoint.x' },
                        { path: 'draftPosts.heroImage.focalPoint.y' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPosts.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listDeletedDraftPosts;
}
exports.listDeletedDraftPosts = listDeletedDraftPosts;
/**
 * Gets a draft post by the provided ID.
 *
 * Uses the provided `draftPostId` to retrieve a draft post.
 */
function getDraftPost(payload) {
    function __getDraftPost({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'GET',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts/{draftPostId}', data: payload, host }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPost.editedDate' },
                        { path: 'draftPost.scheduledPublishDate' },
                        { path: 'draftPost.firstPublishedDate' },
                        { path: 'draftPost.createdDate' },
                        { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPost.heroImage.urlExpirationDate' },
                        { path: 'draftPost.richContent.metadata.createdTimestamp' },
                        { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPost.moderationDetails.submittedDate' },
                        { path: 'draftPost.moderationDetails.moderationDate' },
                        { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPost.coverMedia.image.focalPoint.x' },
                        { path: 'draftPost.coverMedia.image.focalPoint.y' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPost.heroImage.focalPoint.x' },
                        { path: 'draftPost.heroImage.focalPoint.y' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPost.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPost.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getDraftPost;
}
exports.getDraftPost = getDraftPost;
/** Updates a draft post. */
function updateDraftPost(payload) {
    function __updateDraftPost({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: field_mask_1.transformSDKFieldMaskToRESTFieldMask,
                paths: [{ path: 'fieldMask' }],
            },
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'scheduledPublishDate' },
                    { path: 'draftPost.editedDate' },
                    { path: 'draftPost.scheduledPublishDate' },
                    { path: 'draftPost.firstPublishedDate' },
                    { path: 'draftPost.createdDate' },
                    { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                    { path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate' },
                    { path: 'draftPost.heroImage.urlExpirationDate' },
                    { path: 'draftPost.richContent.metadata.createdTimestamp' },
                    { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                    { path: 'draftPost.moderationDetails.submittedDate' },
                    { path: 'draftPost.moderationDetails.moderationDate' },
                    { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                    { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                    },
                ],
            },
            {
                transformFn: float_1.transformSDKFloatToRESTFloat,
                paths: [
                    { path: 'draftPost.coverMedia.image.focalPoint.x' },
                    { path: 'draftPost.coverMedia.image.focalPoint.y' },
                    { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                    { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                    { path: 'draftPost.heroImage.focalPoint.x' },
                    { path: 'draftPost.heroImage.focalPoint.y' },
                    { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                    { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                    },
                    {
                        path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                    },
                    { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                    { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                    },
                    { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                    { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                    { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                    { path: 'draftPost.richContent.nodes.pollData.poll.image.duration' },
                    {
                        path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                    },
                    {
                        path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                    },
                    { path: 'draftPost.richContent.nodes.appEmbedData.image.duration' },
                    { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                    { path: 'draftPost.richContent.nodes.videoData.thumbnail.duration' },
                    { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                    { path: 'draftPost.richContent.nodes.audioData.coverImage.duration' },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'PATCH',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.UpdateDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({
                protoPath: '/v3/draft-posts/{draftPost.id}',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPost.editedDate' },
                        { path: 'draftPost.scheduledPublishDate' },
                        { path: 'draftPost.firstPublishedDate' },
                        { path: 'draftPost.createdDate' },
                        { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPost.heroImage.urlExpirationDate' },
                        { path: 'draftPost.richContent.metadata.createdTimestamp' },
                        { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPost.moderationDetails.submittedDate' },
                        { path: 'draftPost.moderationDetails.moderationDate' },
                        { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPost.coverMedia.image.focalPoint.x' },
                        { path: 'draftPost.coverMedia.image.focalPoint.y' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPost.heroImage.focalPoint.x' },
                        { path: 'draftPost.heroImage.focalPoint.y' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPost.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPost.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __updateDraftPost;
}
exports.updateDraftPost = updateDraftPost;
/**
 * Moves a draft post with the provided ID to the trash bin.
 * A published post can also be deleted by its provided `draftPostId`.
 *
 * The optional `permanent` field enables you to delete a post permanently, bypassing the trash bin. When a draft post is deleted this way, it can't be restored.
 */
function deleteDraftPost(payload) {
    function __deleteDraftPost({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'DELETE',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.DeleteDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts/{draftPostId}', data: payload, host }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __deleteDraftPost;
}
exports.deleteDraftPost = deleteDraftPost;
/**
 * Permanently deletes a draft post by the provided ID from the trash bin.
 *
 * Uses the provided `draftPostId` to permanently delete a draft post from the trash bin. This action is permanent and cannot be reversed.
 */
function removeFromTrashBin(payload) {
    function __removeFromTrashBin({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'DELETE',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RemoveFromTrashBin',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({
                protoPath: '/v3/draft-posts/trash-bin/{draftPostId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
        };
        return metadata;
    }
    return __removeFromTrashBin;
}
exports.removeFromTrashBin = removeFromTrashBin;
/** Deletes multiple draft posts. */
function bulkDeleteDraftPosts(payload) {
    function __bulkDeleteDraftPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'DELETE',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkDeleteDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/bulk/draft-posts', data: payload, host }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'results.item.editedDate' },
                        { path: 'results.item.scheduledPublishDate' },
                        { path: 'results.item.firstPublishedDate' },
                        { path: 'results.item.createdDate' },
                        { path: 'results.item.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'results.item.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'results.item.heroImage.urlExpirationDate' },
                        { path: 'results.item.richContent.metadata.createdTimestamp' },
                        { path: 'results.item.richContent.metadata.updatedTimestamp' },
                        { path: 'results.item.moderationDetails.submittedDate' },
                        { path: 'results.item.moderationDetails.moderationDate' },
                        { path: 'results.item.media.wixMedia.image.urlExpirationDate' },
                        { path: 'results.item.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'results.item.coverMedia.image.focalPoint.x' },
                        { path: 'results.item.coverMedia.image.focalPoint.y' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'results.item.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'results.item.heroImage.focalPoint.x' },
                        { path: 'results.item.heroImage.focalPoint.y' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.x' },
                        { path: 'results.item.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'results.item.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.galleryData.options.item.ratio',
                        },
                        {
                            path: 'results.item.richContent.nodes.imageData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lat',
                        },
                        {
                            path: 'results.item.richContent.nodes.mapData.mapSettings.lng',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.appEmbedData.image.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.video.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.videoData.thumbnail.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.audio.duration',
                        },
                        {
                            path: 'results.item.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __bulkDeleteDraftPosts;
}
exports.bulkDeleteDraftPosts = bulkDeleteDraftPosts;
/**
 * Retrieves a list of up to 100 draft posts per request.
 *
 * List Draft Posts runs with these defaults, which you can override:
 * - `editedDate` is sorted in descending order.
 * - `paging.limit` is `50`.
 * - `paging.offset` is `0`.
 */
function listDraftPosts(payload) {
    function __listDraftPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'GET',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts', data: payload, host }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPosts.editedDate' },
                        { path: 'draftPosts.scheduledPublishDate' },
                        { path: 'draftPosts.firstPublishedDate' },
                        { path: 'draftPosts.createdDate' },
                        { path: 'draftPosts.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPosts.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPosts.heroImage.urlExpirationDate' },
                        { path: 'draftPosts.richContent.metadata.createdTimestamp' },
                        { path: 'draftPosts.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPosts.moderationDetails.submittedDate' },
                        { path: 'draftPosts.moderationDetails.moderationDate' },
                        { path: 'draftPosts.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPosts.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPosts.coverMedia.image.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.image.focalPoint.y' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPosts.heroImage.focalPoint.x' },
                        { path: 'draftPosts.heroImage.focalPoint.y' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPosts.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listDraftPosts;
}
exports.listDraftPosts = listDraftPosts;
/**
 * Gets a deleted draft post from the trash bin by the provided ID.
 *
 * Uses the provided `draftPostId` to retrieve a previously deleted draft post from the trash bin.
 */
function getDeletedDraftPost(payload) {
    function __getDeletedDraftPost({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'GET',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDeletedDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({
                protoPath: '/v3/draft-posts/trash-bin/{draftPostId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPost.editedDate' },
                        { path: 'draftPost.scheduledPublishDate' },
                        { path: 'draftPost.firstPublishedDate' },
                        { path: 'draftPost.createdDate' },
                        { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPost.heroImage.urlExpirationDate' },
                        { path: 'draftPost.richContent.metadata.createdTimestamp' },
                        { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPost.moderationDetails.submittedDate' },
                        { path: 'draftPost.moderationDetails.moderationDate' },
                        { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPost.coverMedia.image.focalPoint.x' },
                        { path: 'draftPost.coverMedia.image.focalPoint.y' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPost.heroImage.focalPoint.x' },
                        { path: 'draftPost.heroImage.focalPoint.y' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPost.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPost.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getDeletedDraftPost;
}
exports.getDeletedDraftPost = getDeletedDraftPost;
/**
 * Restores a deleted draft post from the trash bin by the provided ID.
 *
 * Uses the `draftPostId` to restore a deleted draft post from the trash bin.
 */
function restoreFromTrashBin(payload) {
    function __restoreFromTrashBin({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'POST',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RestoreFromTrashBin',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({
                protoPath: '/v3/draft-posts/trash-bin/{draftPostId}/restore',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPost.editedDate' },
                        { path: 'draftPost.scheduledPublishDate' },
                        { path: 'draftPost.firstPublishedDate' },
                        { path: 'draftPost.createdDate' },
                        { path: 'draftPost.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPost.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPost.heroImage.urlExpirationDate' },
                        { path: 'draftPost.richContent.metadata.createdTimestamp' },
                        { path: 'draftPost.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPost.moderationDetails.submittedDate' },
                        { path: 'draftPost.moderationDetails.moderationDate' },
                        { path: 'draftPost.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPost.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPost.coverMedia.image.focalPoint.x' },
                        { path: 'draftPost.coverMedia.image.focalPoint.y' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPost.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPost.heroImage.focalPoint.x' },
                        { path: 'draftPost.heroImage.focalPoint.y' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.x' },
                        { path: 'draftPost.media.wixMedia.videoV2.posters.focalPoint.y' },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPost.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPost.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPost.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPost.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPost.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPost.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPost.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __restoreFromTrashBin;
}
exports.restoreFromTrashBin = restoreFromTrashBin;
/**
 * Retrieves a list of up to 100 draft posts, given the provided paging, filtering, and sorting.
 *
 * Query Draft Posts runs with these defaults, which you can override:
 * - `editedDate` is sorted in `DESC` order.
 * - `paging.limit` is `50`.
 * - `paging.offset` is `0`.
 */
function queryDraftPosts(payload) {
    function __queryDraftPosts({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'POST',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.QueryDraftPosts',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({ protoPath: '/v3/draft-posts/query', data: payload, host }),
            data: payload,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'draftPosts.editedDate' },
                        { path: 'draftPosts.scheduledPublishDate' },
                        { path: 'draftPosts.firstPublishedDate' },
                        { path: 'draftPosts.createdDate' },
                        { path: 'draftPosts.coverMedia.image.urlExpirationDate' },
                        {
                            path: 'draftPosts.coverMedia.video.thumbnail.urlExpirationDate',
                        },
                        { path: 'draftPosts.heroImage.urlExpirationDate' },
                        { path: 'draftPosts.richContent.metadata.createdTimestamp' },
                        { path: 'draftPosts.richContent.metadata.updatedTimestamp' },
                        { path: 'draftPosts.moderationDetails.submittedDate' },
                        { path: 'draftPosts.moderationDetails.moderationDate' },
                        { path: 'draftPosts.media.wixMedia.image.urlExpirationDate' },
                        { path: 'draftPosts.media.wixMedia.videoV2.urlExpirationDate' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate',
                        },
                    ],
                },
                {
                    transformFn: float_2.transformRESTFloatToSDKFloat,
                    paths: [
                        { path: 'draftPosts.coverMedia.image.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.image.focalPoint.y' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.x' },
                        { path: 'draftPosts.coverMedia.video.thumbnail.focalPoint.y' },
                        { path: 'draftPosts.heroImage.focalPoint.x' },
                        { path: 'draftPosts.heroImage.focalPoint.y' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.x' },
                        { path: 'draftPosts.media.wixMedia.image.focalPoint.y' },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.x',
                        },
                        {
                            path: 'draftPosts.media.wixMedia.videoV2.posters.focalPoint.y',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.image.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.media.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.galleryData.options.item.ratio',
                        },
                        { path: 'draftPosts.richContent.nodes.imageData.image.duration' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lat' },
                        { path: 'draftPosts.richContent.nodes.mapData.mapSettings.lng' },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.poll.options.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.pollData.design.poll.background.image.duration',
                        },
                        {
                            path: 'draftPosts.richContent.nodes.appEmbedData.image.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.videoData.video.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.videoData.thumbnail.duration',
                        },
                        { path: 'draftPosts.richContent.nodes.audioData.audio.duration' },
                        {
                            path: 'draftPosts.richContent.nodes.audioData.coverImage.duration',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __queryDraftPosts;
}
exports.queryDraftPosts = queryDraftPosts;
/**
 * Publishes a specified draft post by ID. This creates a new post entity with the data from the draft post.
 *
 * If the specified draft post was already published, the published post will be updated with the latest values from the draft post entity.
 */
function publishDraftPost(payload) {
    function __publishDraftPost({ host }) {
        const metadata = {
            entityFqdn: 'wix.blog.v3.draft',
            method: 'POST',
            methodFqn: 'com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.PublishDraftPost',
            packageName: PACKAGE_NAME,
            url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl({
                protoPath: '/v3/draft-posts/{draftPostId}/publish',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __publishDraftPost;
}
exports.publishDraftPost = publishDraftPost;
