// @wix/auto_sdk_blog_categories/build/internal/es/src/blog-v3-category-categories.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTImageToSDKImage } from '@wix/sdk-runtime/transformations/image';
import { transformRESTPageURLToSDKPageURL } from '@wix/sdk-runtime/transformations/page-url';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { createCategory as universalCreateCategory, deleteCategory as universalDeleteCategory, getCategory as universalGetCategory, getCategoryBySlug as universalGetCategoryBySlug, listCategories as universalListCategories, queryCategories as universalQueryCategories, updateCategory as universalUpdateCategory, } from './blog-v3-category-categories.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/blog' };
export function createCategory(httpClient) {
    return (category, options) => universalCreateCategory(category, options, 
    // @ts-ignore
    { httpClient });
}
export function updateCategory(httpClient) {
    return (_id, category, options) => universalUpdateCategory(_id, category, options, 
    // @ts-ignore
    { httpClient });
}
export function getCategory(httpClient) {
    return (categoryId, options) => universalGetCategory(categoryId, options, 
    // @ts-ignore
    { httpClient });
}
export function getCategoryBySlug(httpClient) {
    return (slug, options) => universalGetCategoryBySlug(slug, options, 
    // @ts-ignore
    { httpClient });
}
export function listCategories(httpClient) {
    return (options) => universalListCategories(options, 
    // @ts-ignore
    { httpClient });
}
export function queryCategories(httpClient) {
    return (options) => universalQueryCategories(options, 
    // @ts-ignore
    { httpClient });
}
export function deleteCategory(httpClient) {
    return (categoryId) => universalDeleteCategory(categoryId, 
    // @ts-ignore
    { httpClient });
}
export const onCategoryCreated = EventDefinition('wix.blog.v3.category_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'entity.coverImage' }],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onCategoryDeleted = EventDefinition('wix.blog.v3.category_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [
            { path: 'undefined.url' },
            { path: 'undefined.translations.url' },
        ],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'undefined.coverImage' }],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onCategoryUpdated = EventDefinition('wix.blog.v3.category_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: transformRESTImageToSDKImage,
        paths: [{ path: 'entity.coverImage' }],
    },
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { Field, SortOrder, WebhookIdentityType, } from './blog-v3-category-categories.universal.js';
