// @wix/auto_sdk_blog_draft-posts/build/cjs/src/blog-v3-draft-draft-posts.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.TotalDraftPostsGroupingField = exports.ThumbnailsAlignment = exports.TextAlignment = exports.Target = exports.Status = exports.Source = exports.SortOrder = exports.PollLayoutType = exports.PollLayoutDirection = exports.PluginContainerDataAlignment = exports.Origin = exports.Orientation = exports.NullValue = exports.NodeType = exports.ModerationStatusStatus = exports.MapType = exports.LineStyle = exports.LayoutType = exports.InitialExpandedItems = exports.GetDraftPostsSort = exports.GIFType = exports.FontType = exports.Field = exports.Direction = exports.DecorationType = exports.Crop = exports.ButtonDataType = exports.BackgroundType = exports.AppType = exports.Alignment = exports.Action = exports.onDraftUpdated = exports.onDraftDeleted = exports.onDraftCreated = exports.publishDraftPost = exports.queryDraftPosts = exports.restoreFromTrashBin = exports.getDeletedDraftPost = exports.listDraftPosts = exports.bulkDeleteDraftPosts = exports.removeFromTrashBin = exports.deleteDraftPost = exports.updateDraftPost = exports.getDraftPost = exports.listDeletedDraftPosts = exports.bulkUpdateDraftPosts = exports.bulkCreateDraftPosts = exports.createDraftPost = exports.__metadata = void 0;
exports.WidthType = exports.Width = exports.WebhookIdentityType = exports.VoteRole = exports.ViewRole = exports.ViewMode = exports.VerticalAlignment = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const float_1 = require("@wix/sdk-runtime/transformations/float");
const image_1 = require("@wix/sdk-runtime/transformations/image");
const page_url_1 = require("@wix/sdk-runtime/transformations/page-url");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const video_1 = require("@wix/sdk-runtime/transformations/video");
const video_v2_1 = require("@wix/sdk-runtime/transformations/video-v2");
const sdk_types_1 = require("@wix/sdk-types");
const blog_v3_draft_draft_posts_universal_js_1 = require("./blog-v3-draft-draft-posts.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function createDraftPost(httpClient) {
    return (draftPost, options) => (0, blog_v3_draft_draft_posts_universal_js_1.createDraftPost)(draftPost, options, 
    // @ts-ignore
    { httpClient });
}
exports.createDraftPost = createDraftPost;
function bulkCreateDraftPosts(httpClient) {
    return (draftPosts, options) => (0, blog_v3_draft_draft_posts_universal_js_1.bulkCreateDraftPosts)(draftPosts, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkCreateDraftPosts = bulkCreateDraftPosts;
function bulkUpdateDraftPosts(httpClient) {
    return (options) => (0, blog_v3_draft_draft_posts_universal_js_1.bulkUpdateDraftPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkUpdateDraftPosts = bulkUpdateDraftPosts;
function listDeletedDraftPosts(httpClient) {
    return (options) => (0, blog_v3_draft_draft_posts_universal_js_1.listDeletedDraftPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listDeletedDraftPosts = listDeletedDraftPosts;
function getDraftPost(httpClient) {
    return (draftPostId, options) => (0, blog_v3_draft_draft_posts_universal_js_1.getDraftPost)(draftPostId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getDraftPost = getDraftPost;
function updateDraftPost(httpClient) {
    return (_id, draftPost, options) => (0, blog_v3_draft_draft_posts_universal_js_1.updateDraftPost)(_id, draftPost, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateDraftPost = updateDraftPost;
function deleteDraftPost(httpClient) {
    return (draftPostId, options) => (0, blog_v3_draft_draft_posts_universal_js_1.deleteDraftPost)(draftPostId, options, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDraftPost = deleteDraftPost;
function removeFromTrashBin(httpClient) {
    return (draftPostId) => (0, blog_v3_draft_draft_posts_universal_js_1.removeFromTrashBin)(draftPostId, 
    // @ts-ignore
    { httpClient });
}
exports.removeFromTrashBin = removeFromTrashBin;
function bulkDeleteDraftPosts(httpClient) {
    return (postIds, options) => (0, blog_v3_draft_draft_posts_universal_js_1.bulkDeleteDraftPosts)(postIds, options, 
    // @ts-ignore
    { httpClient });
}
exports.bulkDeleteDraftPosts = bulkDeleteDraftPosts;
function listDraftPosts(httpClient) {
    return (options) => (0, blog_v3_draft_draft_posts_universal_js_1.listDraftPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listDraftPosts = listDraftPosts;
function getDeletedDraftPost(httpClient) {
    return (draftPostId) => (0, blog_v3_draft_draft_posts_universal_js_1.getDeletedDraftPost)(draftPostId, 
    // @ts-ignore
    { httpClient });
}
exports.getDeletedDraftPost = getDeletedDraftPost;
function restoreFromTrashBin(httpClient) {
    return (draftPostId) => (0, blog_v3_draft_draft_posts_universal_js_1.restoreFromTrashBin)(draftPostId, 
    // @ts-ignore
    { httpClient });
}
exports.restoreFromTrashBin = restoreFromTrashBin;
function queryDraftPosts(httpClient) {
    return (options) => (0, blog_v3_draft_draft_posts_universal_js_1.queryDraftPosts)(options, 
    // @ts-ignore
    { httpClient });
}
exports.queryDraftPosts = queryDraftPosts;
function publishDraftPost(httpClient) {
    return (draftPostId) => (0, blog_v3_draft_draft_posts_universal_js_1.publishDraftPost)(draftPostId, 
    // @ts-ignore
    { httpClient });
}
exports.publishDraftPost = publishDraftPost;
exports.onDraftCreated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.draft_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
exports.onDraftDeleted = (0, sdk_types_1.EventDefinition)('wix.blog.v3.draft_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
exports.onDraftUpdated = (0, sdk_types_1.EventDefinition)('wix.blog.v3.draft_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: image_1.transformRESTImageToSDKImage,
        paths: [
            { path: 'entity.heroImage' },
            { path: 'entity.coverMedia.image' },
            { path: 'entity.media.wixMedia.image' },
        ],
    },
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.editedDate' },
            { path: 'entity.scheduledPublishDate' },
            { path: 'entity.firstPublishedDate' },
            { path: 'entity.createdDate' },
            { path: 'entity.richContent.metadata.createdTimestamp' },
            { path: 'entity.richContent.metadata.updatedTimestamp' },
            { path: 'entity.moderationDetails.submittedDate' },
            { path: 'entity.moderationDetails.moderationDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: page_url_1.transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }, { path: 'entity.translations.url' }],
    },
    {
        transformFn: video_1.transformRESTVideoToSDKVideo,
        paths: [{ path: 'entity.coverMedia.video' }],
    },
    {
        transformFn: float_1.transformRESTFloatToSDKFloat,
        paths: [
            {
                path: 'entity.richContent.nodes.galleryData.items.image.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.media.duration',
            },
            {
                path: 'entity.richContent.nodes.galleryData.items.video.thumbnail.duration',
            },
            { path: 'entity.richContent.nodes.galleryData.options.item.ratio' },
            { path: 'entity.richContent.nodes.imageData.image.duration' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lat' },
            { path: 'entity.richContent.nodes.mapData.mapSettings.lng' },
            { path: 'entity.richContent.nodes.pollData.poll.image.duration' },
            {
                path: 'entity.richContent.nodes.pollData.poll.options.image.duration',
            },
            {
                path: 'entity.richContent.nodes.pollData.design.poll.background.image.duration',
            },
            { path: 'entity.richContent.nodes.appEmbedData.image.duration' },
            { path: 'entity.richContent.nodes.videoData.video.duration' },
            { path: 'entity.richContent.nodes.videoData.thumbnail.duration' },
            { path: 'entity.richContent.nodes.audioData.audio.duration' },
            { path: 'entity.richContent.nodes.audioData.coverImage.duration' },
        ],
    },
    {
        transformFn: video_v2_1.transformRESTVideoV2ToSDKVideoV2,
        paths: [{ path: 'entity.media.wixMedia.videoV2' }],
    },
])))();
var blog_v3_draft_draft_posts_universal_js_2 = require("./blog-v3-draft-draft-posts.universal.js");
Object.defineProperty(exports, "Action", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Action; } });
Object.defineProperty(exports, "Alignment", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Alignment; } });
Object.defineProperty(exports, "AppType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.AppType; } });
Object.defineProperty(exports, "BackgroundType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.BackgroundType; } });
Object.defineProperty(exports, "ButtonDataType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.ButtonDataType; } });
Object.defineProperty(exports, "Crop", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Crop; } });
Object.defineProperty(exports, "DecorationType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.DecorationType; } });
Object.defineProperty(exports, "Direction", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Direction; } });
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Field; } });
Object.defineProperty(exports, "FontType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.FontType; } });
Object.defineProperty(exports, "GIFType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.GIFType; } });
Object.defineProperty(exports, "GetDraftPostsSort", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.GetDraftPostsSort; } });
Object.defineProperty(exports, "InitialExpandedItems", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.InitialExpandedItems; } });
Object.defineProperty(exports, "LayoutType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.LayoutType; } });
Object.defineProperty(exports, "LineStyle", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.LineStyle; } });
Object.defineProperty(exports, "MapType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.MapType; } });
Object.defineProperty(exports, "ModerationStatusStatus", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.ModerationStatusStatus; } });
Object.defineProperty(exports, "NodeType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.NodeType; } });
Object.defineProperty(exports, "NullValue", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.NullValue; } });
Object.defineProperty(exports, "Orientation", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Orientation; } });
Object.defineProperty(exports, "Origin", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Origin; } });
Object.defineProperty(exports, "PluginContainerDataAlignment", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.PluginContainerDataAlignment; } });
Object.defineProperty(exports, "PollLayoutDirection", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.PollLayoutDirection; } });
Object.defineProperty(exports, "PollLayoutType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.PollLayoutType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Source", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Source; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Status; } });
Object.defineProperty(exports, "Target", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Target; } });
Object.defineProperty(exports, "TextAlignment", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.TextAlignment; } });
Object.defineProperty(exports, "ThumbnailsAlignment", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.ThumbnailsAlignment; } });
Object.defineProperty(exports, "TotalDraftPostsGroupingField", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.TotalDraftPostsGroupingField; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Type; } });
Object.defineProperty(exports, "VerticalAlignment", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.VerticalAlignment; } });
Object.defineProperty(exports, "ViewMode", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.ViewMode; } });
Object.defineProperty(exports, "ViewRole", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.ViewRole; } });
Object.defineProperty(exports, "VoteRole", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.VoteRole; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "Width", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.Width; } });
Object.defineProperty(exports, "WidthType", { enumerable: true, get: function () { return blog_v3_draft_draft_posts_universal_js_2.WidthType; } });
