// @wix/auto_sdk_redirects_redirects/build/internal/es/src/headless-v1-redirect-session-redirects.types.js
export var LocationType;
(function (LocationType) {
    LocationType["UNDEFINED"] = "UNDEFINED";
    LocationType["OWNER_BUSINESS"] = "OWNER_BUSINESS";
    LocationType["OWNER_CUSTOM"] = "OWNER_CUSTOM";
    LocationType["CUSTOM"] = "CUSTOM";
})(LocationType || (LocationType = {}));
/** Specifies the behavior of the authorization flow regarding site member authentication and consent. */
export var Prompt;
(function (Prompt) {
    /** Display the login page to the member, prompting them to enter their credentials. */
    Prompt["login"] = "login";
    /**
     * Bypass the Wix login page and redirect the member directly to your site.
     * This option checks if the member has an active session from a previous login
     * and, if so, immediately redirects to the specified URL while retrieving tokens.
     */
    Prompt["none"] = "none";
})(Prompt || (Prompt = {}));
export var MembersAccountSection;
(function (MembersAccountSection) {
    /** Account info section in "my account". */
    MembersAccountSection["ACCOUNT_INFO"] = "ACCOUNT_INFO";
    /** My Bookings section in "my account". */
    MembersAccountSection["BOOKINGS"] = "BOOKINGS";
    /** My Orders section in "my account". */
    MembersAccountSection["ORDERS"] = "ORDERS";
    /** Subscriptions section in "my account". */
    MembersAccountSection["SUBSCRIPTIONS"] = "SUBSCRIPTIONS";
    /** Events section in "my account". */
    MembersAccountSection["EVENTS"] = "EVENTS";
})(MembersAccountSection || (MembersAccountSection = {}));
export var AttachPagesResponseStatus;
(function (AttachPagesResponseStatus) {
    /** Invalid value. */
    AttachPagesResponseStatus["UNKNOWN"] = "UNKNOWN";
    /** Pages were successfully attached. */
    AttachPagesResponseStatus["SUCCESS"] = "SUCCESS";
    /** No pages were attached because the site is already published. */
    AttachPagesResponseStatus["NO_ACTION"] = "NO_ACTION";
    /** An error occurred, such as when the site or app is not found. */
    AttachPagesResponseStatus["ERROR"] = "ERROR";
})(AttachPagesResponseStatus || (AttachPagesResponseStatus = {}));
export var CallbackType;
(function (CallbackType) {
    /** Invalid value. */
    CallbackType["UNKNOWN"] = "UNKNOWN";
    /** Callback URL is used for the logout flow. */
    CallbackType["LOGOUT"] = "LOGOUT";
    /** Callback URL is used for a checkout flow. */
    CallbackType["CHECKOUT"] = "CHECKOUT";
    /** Callback URL is used for the authorize flow. */
    CallbackType["AUTHORIZE"] = "AUTHORIZE";
})(CallbackType || (CallbackType = {}));
export var Status;
(function (Status) {
    /** Invalid value. */
    Status["UNKNOWN"] = "UNKNOWN";
    /** Templates were successfully attached. */
    Status["SUCCESS"] = "SUCCESS";
    /** An error occurred. */
    Status["ERROR"] = "ERROR";
})(Status || (Status = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
