// @wix/auto_sdk_blog_draft-posts@1.0.2/build/es/src/blog-v3-draft-draft-posts.meta.js
import * as ambassadorWixBlogV3Draft from './blog-v3-draft-draft-posts.http.js';
export function createDraftPost() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.createDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkCreateDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkCreateDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/draft-posts/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkUpdateDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkUpdateDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/draft-posts/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDeletedDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.listDeletedDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/trash-bin',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.getDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDraftPost() {
    const payload = { draftPost: { id: ':draftPostId' } };
    const getRequestOptions = ambassadorWixBlogV3Draft.updateDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/draft-posts/{draftPost.id}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.deleteDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/draft-posts/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeFromTrashBin() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.removeFromTrashBin(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/draft-posts/trash-bin/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function bulkDeleteDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkDeleteDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/bulk/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.listDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDeletedDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.getDeletedDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/trash-bin/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function restoreFromTrashBin() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.restoreFromTrashBin(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/trash-bin/{draftPostId}/restore',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.queryDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function publishDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.publishDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/{draftPostId}/publish',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
