// @wix/auto_sdk_blog_draft-posts@1.0.2/build/internal/cjs/src/blog-v3-draft-draft-posts.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.publishDraftPost = exports.queryDraftPosts = exports.restoreFromTrashBin = exports.getDeletedDraftPost = exports.listDraftPosts = exports.bulkDeleteDraftPosts = exports.removeFromTrashBin = exports.deleteDraftPost = exports.updateDraftPost = exports.getDraftPost = exports.listDeletedDraftPosts = exports.bulkUpdateDraftPosts = exports.bulkCreateDraftPosts = exports.createDraftPost = void 0;
const ambassadorWixBlogV3Draft = __importStar(require("./blog-v3-draft-draft-posts.http.js"));
function createDraftPost() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.createDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDraftPost = createDraftPost;
function bulkCreateDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkCreateDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/bulk/draft-posts/create',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkCreateDraftPosts = bulkCreateDraftPosts;
function bulkUpdateDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkUpdateDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/draft-posts/update',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkUpdateDraftPosts = bulkUpdateDraftPosts;
function listDeletedDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.listDeletedDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/trash-bin',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDeletedDraftPosts = listDeletedDraftPosts;
function getDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.getDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDraftPost = getDraftPost;
function updateDraftPost() {
    const payload = { draftPost: { id: ':draftPostId' } };
    const getRequestOptions = ambassadorWixBlogV3Draft.updateDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PATCH',
        path: '/v3/draft-posts/{draftPost.id}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDraftPost = updateDraftPost;
function deleteDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.deleteDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/draft-posts/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDraftPost = deleteDraftPost;
function removeFromTrashBin() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.removeFromTrashBin(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/draft-posts/trash-bin/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeFromTrashBin = removeFromTrashBin;
function bulkDeleteDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.bulkDeleteDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/bulk/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.bulkDeleteDraftPosts = bulkDeleteDraftPosts;
function listDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.listDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDraftPosts = listDraftPosts;
function getDeletedDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.getDeletedDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/draft-posts/trash-bin/{draftPostId}',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDeletedDraftPost = getDeletedDraftPost;
function restoreFromTrashBin() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.restoreFromTrashBin(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/trash-bin/{draftPostId}/restore',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.restoreFromTrashBin = restoreFromTrashBin;
function queryDraftPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Draft.queryDraftPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.queryDraftPosts = queryDraftPosts;
function publishDraftPost() {
    const payload = { draftPostId: ':draftPostId' };
    const getRequestOptions = ambassadorWixBlogV3Draft.publishDraftPost(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/draft-posts/{draftPostId}/publish',
        pathParams: { draftPostId: 'draftPostId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.publishDraftPost = publishDraftPost;
