// @wix/auto_sdk_blog_blog-importer@1.0.0/build/internal/es/src/blog-v1-wordpress-import-blog-importer.meta.js
import * as ambassadorWixBlogV1WordpressImport from './blog-v1-wordpress-import-blog-importer.http.js';
export function startImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.startImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function submitUrlForImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.submitUrlForImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/prepare-wordpress-url-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function validateUrlForImport() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.validateUrlForImport(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/validate-wordpress-url-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteNotImported() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.deleteNotImported(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getImportStatus() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.getImportStatus(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/wordpress-import',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getNotImportedPosts() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV1WordpressImport.getNotImportedPosts(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/wordpress-import/posts/not-imported',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
