// @wix/auto_sdk_blog_categories@1.0.1/build/es/src/blog-v3-category-categories.universal.js
import { transformError as sdkTransformError } from '@wix/sdk-runtime/transform-error';
import { queryBuilder } from '@wix/sdk-runtime/query-builder';
import { renameKeysFromSDKRequestToRESTRequest, renameKeysFromRESTResponseToSDKResponse, } from '@wix/sdk-runtime/rename-all-nested-keys';
import * as ambassadorWixBlogV3Category from './blog-v3-category-categories.http.js';
// @ts-ignore
import { transformSDKImageToRESTImage } from '@wix/sdk-runtime/transformations/image';
import { transformRESTImageToSDKImage } from '@wix/sdk-runtime/transformations/image';
import { transformSDKPageURLToRESTPageURL } from '@wix/sdk-runtime/transformations/page-url';
import { transformRESTPageURLToSDKPageURL } from '@wix/sdk-runtime/transformations/page-url';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
export var Field;
(function (Field) {
    Field["UNKNOWN"] = "UNKNOWN";
    /** Includes Category url. */
    Field["URL"] = "URL";
    /**
     * Includes internal id field.
     * Reserved for internal use
     */
    Field["INTERNAL_ID"] = "INTERNAL_ID";
    /** Includes SEO data. */
    Field["SEO"] = "SEO";
})(Field || (Field = {}));
export var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (SortOrder = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
/**
 * Creates a category.
 * @param category - Category info.
 * @public
 * @requiredField category
 * @requiredField category.label
 * @permissionId BLOG.CREATE-CATEGORY
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @applicableIdentity APP
 * @returns Category info.
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.CreateCategory
 */
export async function createCategory(category, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = transformPaths(renameKeysFromSDKRequestToRESTRequest({
        category: category,
        fieldsets: options?.fieldsets,
    }), [
        {
            transformFn: transformSDKPageURLToRESTPageURL,
            paths: [
                { path: 'category.url' },
                { path: 'category.translations.url' },
            ],
        },
        {
            transformFn: transformSDKImageToRESTImage,
            paths: [{ path: 'category.coverImage' }],
        },
    ]);
    const reqOpts = ambassadorWixBlogV3Category.createCategory(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTPageURLToSDKPageURL,
                paths: [
                    { path: 'category.url' },
                    { path: 'category.translations.url' },
                ],
            },
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'category.coverImage' }],
            },
        ]))?.category;
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                category: '$[0]',
                fieldsets: '$[1].fieldsets',
            },
            singleArgumentUnchanged: false,
        }, ['category', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Updates a category.
 * @param _id - Category ID.
 * @public
 * @requiredField _id
 * @requiredField category
 * @permissionId BLOG.EDIT-CATEGORY
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @applicableIdentity APP
 * @returns Category info.
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.UpdateCategory
 */
export async function updateCategory(_id, category, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[3];
    const payload = transformPaths(renameKeysFromSDKRequestToRESTRequest({
        category: { ...category, id: _id },
        fieldsets: options?.fieldsets,
    }), [
        {
            transformFn: transformSDKPageURLToRESTPageURL,
            paths: [
                { path: 'category.url' },
                { path: 'category.translations.url' },
            ],
        },
        {
            transformFn: transformSDKImageToRESTImage,
            paths: [{ path: 'category.coverImage' }],
        },
    ]);
    const reqOpts = ambassadorWixBlogV3Category.updateCategory(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTPageURLToSDKPageURL,
                paths: [
                    { path: 'category.url' },
                    { path: 'category.translations.url' },
                ],
            },
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'category.coverImage' }],
            },
        ]))?.category;
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: { category: '$[1]' },
            explicitPathsToArguments: {
                'category.id': '$[0]',
                fieldsets: '$[2].fieldsets',
            },
            singleArgumentUnchanged: false,
        }, ['_id', 'category', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Gets a category with the specified ID.
 * @param categoryId - Category ID.
 * @public
 * @requiredField categoryId
 * @param options - Options specifying which fields to return.
 * @permissionId BLOG.READ-CATEGORY
 * @permissionScope Read Blog
 * @permissionScopeId SCOPE.DC-BLOG.READ-BLOGS
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @permissionScope Read Draft Blog Posts
 * @permissionScopeId SCOPE.DC-BLOG.READ-DRAFT-POSTS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.GetCategory
 */
export async function getCategory(categoryId, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        categoryId: categoryId,
        fieldsets: options?.fieldsets,
    });
    const reqOpts = ambassadorWixBlogV3Category.getCategory(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTPageURLToSDKPageURL,
                paths: [
                    { path: 'category.url' },
                    { path: 'category.translations.url' },
                ],
            },
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'category.coverImage' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                categoryId: '$[0]',
                fieldsets: '$[1].fieldsets',
            },
            singleArgumentUnchanged: false,
        }, ['categoryId', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Gets a category with the specified slug.
 *
 * The slug is at the end of the URL of a specific category.
 * For example, if a category's URL is `https://example.com/blog/categories/famous-cats`, the slug is `famous-cats`.
 * The slug is a case-sensitive string that is derived from the category's label, unless specified otherwise.
 * @public
 * @requiredField slug
 * @param slug - Slug of the category to retrieve.
 *
 * The end of a category's URL. For example, `'https:/example.com/blog/category/{my-category-slug}'`. Case sensitive and generally based on the category `label` if not specified.
 * @param options - Options specifying which fields to return.
 * @permissionId BLOG.READ-CATEGORY
 * @permissionScope Read Blog
 * @permissionScopeId SCOPE.DC-BLOG.READ-BLOGS
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @permissionScope Read Draft Blog Posts
 * @permissionScopeId SCOPE.DC-BLOG.READ-DRAFT-POSTS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.GetCategoryBySlug
 */
export async function getCategoryBySlug(slug, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        slug: slug,
        fieldsets: options?.fieldsets,
        language: options?.language,
    });
    const reqOpts = ambassadorWixBlogV3Category.getCategoryBySlug(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTPageURLToSDKPageURL,
                paths: [
                    { path: 'category.url' },
                    { path: 'category.translations.url' },
                ],
            },
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'category.coverImage' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                slug: '$[0]',
                fieldsets: '$[1].fieldsets',
                language: '$[1].language',
            },
            singleArgumentUnchanged: false,
        }, ['slug', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Retrieves a list of up to 100 categories per request.
 *
 * If you pass a language filter to the method, it returns a list of up to 100 categories in the requested language.
 *
 * The categoies are displayed in order of their `displayPosition`, starting with `0`. The `displayPosition` is the position in which the categories
 * are displayed in the Category Menu page. By default, categories get added to the bottom of the Category Menu page with a `displayPosition` of `-1`.
 *
 * List Categories runs with these defaults, which you can override:
 * - `paging.limit` is `50`.
 * - `paging.offset` is `0`.
 *
 * List Categories is sorted by `displayPosition` in descending order. This cannot be overridden.
 * @public
 * @param options - Filter and paging options.
 * @permissionId BLOG.READ-CATEGORY
 * @permissionScope Read Blog
 * @permissionScopeId SCOPE.DC-BLOG.READ-BLOGS
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @permissionScope Read Draft Blog Posts
 * @permissionScopeId SCOPE.DC-BLOG.READ-DRAFT-POSTS
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.ListCategories
 */
export async function listCategories(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        paging: options?.paging,
        fieldsets: options?.fieldsets,
        language: options?.language,
    });
    const reqOpts = ambassadorWixBlogV3Category.listCategories(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return renameKeysFromRESTResponseToSDKResponse(transformPaths(result.data, [
            {
                transformFn: transformRESTPageURLToSDKPageURL,
                paths: [
                    { path: 'categories.url' },
                    { path: 'categories.translations.url' },
                ],
            },
            {
                transformFn: transformRESTImageToSDKImage,
                paths: [{ path: 'categories.coverImage' }],
            },
        ]));
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                paging: '$[0].paging',
                fieldsets: '$[0].fieldsets',
                language: '$[0].language',
            },
            singleArgumentUnchanged: false,
        }, ['options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
/**
 * Creates a query to retrieve a list of categories.
 *
 *
 * The Query Categories method builds a query to retrieve a list of up to 100 categories per language, and returns a [`CategoriesQueryBuilder`](https://dev.wix.com/docs/sdk/backend-modules/blog/categories/categories-query-builder/ascending) object.
 *
 * The response contains the query definition, which is typically used to run the query using the [`find()`](https://dev.wix.com/docs/sdk/backend-modules/blog/categories/categories-query-builder/find) method.
 *
 * You can refine the query by chaining `CategoriesQueryBuilder` methods to the query. `CategoriesQueryBuilder` methods enable you to sort, filter, and control the results that Query Categories returns. Any methods chained to the Query Categories method are applied in the order that they are called.
 *
 * Query Categories runs with these `CategoriesQueryBuilder` defaults, which you can override.
 * - [`limit(100)`](https://dev.wix.com/docs/sdk/backend-modules/blog/categories/categories-query-builder/limit)
 * - [`ascending(displayPosition)`](https://dev.wix.com/docs/sdk/backend-modules/blog/categories/categories-query-builder/ascending)
 *
 * The following `CategoriesQueryBuilder` methods are supported for Query Categories. For a full description of the `Categories` object, see the response for the [`items`](https://dev.wix.com/docs/sdk/backend-modules/blog/categories/categories-query-result/items) property in [`CategoriesQueryResult`](https://www.wix.com/velo/reference/wix-blog-backend/categories/categoriesqueryresult).
 * @public
 * @param options - Options specifying which fields to return.
 * @permissionScope Read Blog
 * @permissionScopeId SCOPE.DC-BLOG.READ-BLOGS
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @permissionScope Read Draft Blog Posts
 * @permissionScopeId SCOPE.DC-BLOG.READ-DRAFT-POSTS
 * @permissionId BLOG.READ-CATEGORY
 * @applicableIdentity APP
 * @applicableIdentity VISITOR
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.QueryCategories
 */
export function queryCategories(options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    return queryBuilder({
        func: async (payload) => {
            const reqOpts = ambassadorWixBlogV3Category.queryCategories({
                ...payload,
                ...(options ?? {}),
            });
            sideEffects?.onSiteCall?.();
            try {
                const result = await httpClient.request(reqOpts);
                sideEffects?.onSuccess?.(result);
                return result;
            }
            catch (err) {
                sideEffects?.onError?.(err);
                throw err;
            }
        },
        requestTransformer: (query) => {
            const args = [query, options];
            return renameKeysFromSDKRequestToRESTRequest({
                ...args?.[1],
                query: args?.[0],
            });
        },
        responseTransformer: ({ data }) => {
            const transformedData = renameKeysFromRESTResponseToSDKResponse(transformPaths(data, [
                {
                    transformFn: transformRESTPageURLToSDKPageURL,
                    paths: [
                        { path: 'categories.url' },
                        { path: 'categories.translations.url' },
                    ],
                },
                {
                    transformFn: transformRESTImageToSDKImage,
                    paths: [{ path: 'categories.coverImage' }],
                },
            ]));
            return {
                items: transformedData?.categories,
                pagingMetadata: transformedData?.pagingMetadata,
            };
        },
        errorTransformer: (err) => {
            const transformedError = sdkTransformError(err, {
                spreadPathsToArguments: {},
                explicitPathsToArguments: { query: '$[0]' },
                singleArgumentUnchanged: false,
            });
            throw transformedError;
        },
        pagingMethod: 'OFFSET',
        transformationPaths: {},
    });
}
/**
 * Deletes a category.
 * @param categoryId - Category ID.
 * @public
 * @requiredField categoryId
 * @permissionId BLOG.DELETE-CATEGORY
 * @permissionScope Manage Blog
 * @permissionScopeId SCOPE.DC-BLOG.MANAGE-BLOG
 * @applicableIdentity APP
 * @fqn com.wixpress.npm.communities.platformized.blog.v3.CategoryService.DeleteCategory
 */
export async function deleteCategory(categoryId) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = renameKeysFromSDKRequestToRESTRequest({
        categoryId: categoryId,
    });
    const reqOpts = ambassadorWixBlogV3Category.deleteCategory(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = sdkTransformError(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { categoryId: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['categoryId']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
