// @wix/auto_sdk_blog_blog-importer/build/es/src/blog-v1-wordpress-import-blog-importer.public.js
import { deleteNotImported as universalDeleteNotImported, getImportStatus as universalGetImportStatus, getNotImportedPosts as universalGetNotImportedPosts, startImport as universalStartImport, submitUrlForImport as universalSubmitUrlForImport, validateUrlForImport as universalValidateUrlForImport, } from './blog-v1-wordpress-import-blog-importer.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/blog' };
export function startImport(httpClient) {
    return (options) => universalStartImport(options, 
    // @ts-ignore
    { httpClient });
}
export function submitUrlForImport(httpClient) {
    return (url, options) => universalSubmitUrlForImport(url, options, 
    // @ts-ignore
    { httpClient });
}
export function validateUrlForImport(httpClient) {
    return (url) => universalValidateUrlForImport(url, 
    // @ts-ignore
    { httpClient });
}
export function deleteNotImported(httpClient) {
    return () => universalDeleteNotImported(
    // @ts-ignore
    { httpClient });
}
export function getImportStatus(httpClient) {
    return () => universalGetImportStatus(
    // @ts-ignore
    { httpClient });
}
export function getNotImportedPosts(httpClient) {
    return () => universalGetNotImportedPosts(
    // @ts-ignore
    { httpClient });
}
export { Status, } from './blog-v1-wordpress-import-blog-importer.universal.js';
