// @wix/auto_sdk_blog_blog-cache/build/internal/cjs/src/blog-v1-blog-cache-blog-cache.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResolutionMethod = exports.PlacementType = exports.LocaleStatus = exports.Flag = exports.DayOfWeek = exports.getBlogCache = exports.__metadata = void 0;
const blog_v1_blog_cache_blog_cache_universal_js_1 = require("./blog-v1-blog-cache-blog-cache.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/blog' };
function getBlogCache(httpClient) {
    return () => (0, blog_v1_blog_cache_blog_cache_universal_js_1.getBlogCache)(
    // @ts-ignore
    { httpClient });
}
exports.getBlogCache = getBlogCache;
var blog_v1_blog_cache_blog_cache_universal_js_2 = require("./blog-v1-blog-cache-blog-cache.universal.js");
Object.defineProperty(exports, "DayOfWeek", { enumerable: true, get: function () { return blog_v1_blog_cache_blog_cache_universal_js_2.DayOfWeek; } });
Object.defineProperty(exports, "Flag", { enumerable: true, get: function () { return blog_v1_blog_cache_blog_cache_universal_js_2.Flag; } });
Object.defineProperty(exports, "LocaleStatus", { enumerable: true, get: function () { return blog_v1_blog_cache_blog_cache_universal_js_2.LocaleStatus; } });
Object.defineProperty(exports, "PlacementType", { enumerable: true, get: function () { return blog_v1_blog_cache_blog_cache_universal_js_2.PlacementType; } });
Object.defineProperty(exports, "ResolutionMethod", { enumerable: true, get: function () { return blog_v1_blog_cache_blog_cache_universal_js_2.ResolutionMethod; } });
