// @wix/auto_sdk_blog_tags/build/cjs/src/blog-v3-tag-tags.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.SortOrder = exports.Field = exports.onTagUpdated = exports.onTagDeleted = exports.onTagCreated = exports.deleteTag = exports.queryTags = exports.getTagBySlug = exports.getTag = exports.getTagByLabel = exports.createTag = void 0;
const blog_v3_tag_tags_public_js_1 = require("./blog-v3-tag-tags.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const blog_v3_tag_tags_public_js_2 = require("./blog-v3-tag-tags.public.js");
const blog_v3_tag_tags_public_js_3 = require("./blog-v3-tag-tags.public.js");
const blog_v3_tag_tags_public_js_4 = require("./blog-v3-tag-tags.public.js");
exports.createTag = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.createTag);
exports.getTagByLabel = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.getTagByLabel);
exports.getTag = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.getTag);
exports.getTagBySlug = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.getTagBySlug);
exports.queryTags = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.queryTags);
exports.deleteTag = (0, rest_modules_1.createRESTModule)(blog_v3_tag_tags_public_js_1.deleteTag);
/**
 * Triggered when a tag is created.
 */
exports.onTagCreated = (0, event_definition_modules_1.createEventModule)(blog_v3_tag_tags_public_js_2.onTagCreated);
/**
 * Triggered when a tag is deleted.
 */
exports.onTagDeleted = (0, event_definition_modules_1.createEventModule)(blog_v3_tag_tags_public_js_3.onTagDeleted);
/**
 * Triggered when a tag is updated.
 */
exports.onTagUpdated = (0, event_definition_modules_1.createEventModule)(blog_v3_tag_tags_public_js_4.onTagUpdated);
var blog_v3_tag_tags_universal_js_1 = require("./blog-v3-tag-tags.universal.js");
Object.defineProperty(exports, "Field", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_1.Field; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_1.SortOrder; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return blog_v3_tag_tags_universal_js_1.WebhookIdentityType; } });
