// @wix/auto_sdk_blog_tags/build/internal/es/src/blog-v3-tag-tags.meta.js
import * as ambassadorWixBlogV3Tag from './blog-v3-tag-tags.http.js';
export function createTag() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Tag.createTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/tags',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTagByLabel() {
    const payload = { label: ':label' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTagByLabel(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/labels/{label}',
        pathParams: { label: 'label' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTag() {
    const payload = { tagId: ':tagId' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTagBySlug() {
    const payload = { slug: ':slug' };
    const getRequestOptions = ambassadorWixBlogV3Tag.getTagBySlug(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v3/tags/slugs/{slug}',
        pathParams: { slug: 'slug' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function queryTags() {
    const payload = {};
    const getRequestOptions = ambassadorWixBlogV3Tag.queryTags(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v3/tags/query',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteTag() {
    const payload = { tagId: ':tagId' };
    const getRequestOptions = ambassadorWixBlogV3Tag.deleteTag(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v3/tags/{tagId}',
        pathParams: { tagId: 'tagId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
