// @wix/auto_sdk_blog_tags/build/internal/es/src/blog-v3-tag-tags.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTPageURLToSDKPageURL } from '@wix/sdk-runtime/transformations/page-url';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { createTag as universalCreateTag, deleteTag as universalDeleteTag, getTag as universalGetTag, getTagByLabel as universalGetTagByLabel, getTagBySlug as universalGetTagBySlug, queryTags as universalQueryTags, } from './blog-v3-tag-tags.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/blog' };
export function createTag(httpClient) {
    return (label, options) => universalCreateTag(label, options, 
    // @ts-ignore
    { httpClient });
}
export function getTagByLabel(httpClient) {
    return (label, options) => universalGetTagByLabel(label, options, 
    // @ts-ignore
    { httpClient });
}
export function getTag(httpClient) {
    return (tagId, options) => universalGetTag(tagId, options, 
    // @ts-ignore
    { httpClient });
}
export function getTagBySlug(httpClient) {
    return (slug, options) => universalGetTagBySlug(slug, options, 
    // @ts-ignore
    { httpClient });
}
export function queryTags(httpClient) {
    return (options) => universalQueryTags(options, 
    // @ts-ignore
    { httpClient });
}
export function deleteTag(httpClient) {
    return (tagId) => universalDeleteTag(tagId, 
    // @ts-ignore
    { httpClient });
}
export const onTagCreated = EventDefinition('wix.blog.v3.tag_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }],
    },
])))();
export const onTagDeleted = EventDefinition('wix.blog.v3.tag_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.createdDate' },
            { path: 'undefined.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'undefined.url' }],
    },
])))();
export const onTagUpdated = EventDefinition('wix.blog.v3.tag_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.createdDate' },
            { path: 'entity.updatedDate' },
            { path: 'metadata.eventTime' },
        ],
    },
    {
        transformFn: transformRESTPageURLToSDKPageURL,
        paths: [{ path: 'entity.url' }],
    },
])))();
export { Field, SortOrder, WebhookIdentityType, } from './blog-v3-tag-tags.universal.js';
